---
title: "Analysis 5: : Morphological differentiation analysis"
author: "Joe Gunn"
date: "2022-07-28"
output: html_document
---

# Project: Population genetic structure and morphological differentiation between Northern Smallmouth Bass (<i>Micropterus dolomieu dolomieu</i>) and Neosho Smallmouth Bass (<i>M. d. velox</i>)
We investigated patterns of genetic (via microsatellites) and morphological diversity, differentiation, and structure across Smallmouth Bass populations (<i>Micropterus dolomieu</i>) in the Central Interior Highlands (CIH) of North America, including the two recognized subspecies: Northern Smallmouth Bass (<i>M. d. dolomieu</i>), which is native to the lower Ozark Highlands, and Neosho Smallmouth Bass (<i>M. d. velox</i>), which is endemic to tributaries of the Arkansas River Basin. We compared three independent combinations of starting parameters in the Bayesian clustering software program STRUCTURE to ascertain a robust picture of hierarchical genetic structure in the CIH. We paired these data with genetic diversity metrics (e.g., observed and expected heterozygosity and allelic richness) to determine relative amounts of variation across geographically separated populations. We then assessed differentiation in five morphometric and meristic traits. We ultimately aimed to validate or amend the taxonomic status of the Smallmouth Bass subspecies by revealing potential ecological and evolutionary divergence between them, with the hope that increased taxonomic resolution would provide insight into the presence and distribution of evolutionary significant and management units for a popular sportfish.

## Specific Aim: Genetic diversity and differentiation analysis
In this analysis, we assess morphological differentiation between Neosho Smallmouth Bass and Northern Smallmouth Bass in the CIH using five morphometrics (traits with continuous distributions; total length, standard length, body depth, head length, and orbital length) and one meristic (discrete count trait; soft dorsal fin rays) and three independently analysis frameworks: 1) individual analyses of covariance (ANCOVAs) on morphometric traits; 2) multiviariate discriminant function analysis (DFA) on morphometric traits; and 3) logistic regression on soft dorsal fin rays. With ANCOVA, we assess differentiation between subsepecies at each trait individually. With DFA, we assess the accuracy with which an individual can be assigned to either subspecies based on composite trait values. With logistic regression, we assess the probability with which a Neosho Smallmouth Bass and Northern Smallmouth Bass possess a specific number of soft dorsal rays.

## Phases of Analysis
### Phase 1: Univariate morphometric analysis with ANCOVA
### Phase 2: Multivariate morphometric analysis with DFA
### Phase 3: Univariate meristic analysis with logistic regression

## Libraries needed for analysis
```{r}
library(tidyverse)
library(cowplot)
library(readxl)
library(pophelper)
```

## PHASE 1: UNIVARIATE MORPHOMETRIC ANALYSIS WITH ANCOVA
In this phase of the analysis, we assess differentiation between Neosho Smallmouth Bass and Northern Smallmouth Bass in five individual morphometric traits (i.e., total length, standard length, body depth, head length, and orbital length; see Analysis 2 in `../02_data_summary_analysis/smb_genetics_data_summary_analysis` for descriptions for each trait) using analysis of covariance.  Thus, we compare each trait between subspecies as they co-vary with total length. We complete this assessment for two subsets of samples: 1) all fish labeled by the subspecies range from which they were collected (Arkansas River Basin, aka Neosho, vs. Missouri or White River drainages, aka Northern), and 2) only fish that assigned as purely (defined as assigning with >90% ancestry membership to any one genetic cluster based on population structure analysis) Neosho or Northern. 

### STEP 1: Read in raw phenotype data and merge with metadata and population structure membership data
In this step, we prepare the data for analysis by merging all phenotype data with both metadata and population structure membersihp data as ascertained in Analysis 3.

#### 1a. Read in and merge phenotype data with metadata; run the Rmd chunk below.

##### Read in phenotype data and merge with metadata:
```{r}
# read in phenotpye data
load("../02_data_summary_analysis/data/phenotype_data.Rda")

# load sample metadata
load("../02_data_summary_analysis/data/metadata.Rda")

# Merge phenotype data and metadata
phenotypes <- merge(metadata,
                    phenotype_data,
                    by = "sample_id")

# Save combined phenotype data and metadata
save(phenotypes, file = "data/phenotypes.Rda")
```

#### 1b. Read in population structure membership data and merge with phenotype data; run the Rmd chunk below.
In this step, we read in the results of structure analysis with default parameters for K=3, which gives membership coefficients for Neosho Smallmouth Bass and Northern Smallmouth Bass, as well as Spotted Bass. We then merge these membership values with the phenotype data.

##### Read in population structure membership data and merge with metadata:
```{r}
# Read in combined-merged Q table for K=3 from default structure analysis
q <- readQ("../03_structure_analysis/data/structure_data/clumpp_data/default_clumpp/full_data/pop_K3/pop_K3-combined-merged.txt") %>%
  as.data.frame()

# load sample metadata
load("../02_data_summary_analysis/data/metadata.Rda")

# load phenotype data
load("data/phenotypes.Rda")

# name columns by predominant cluster membership 
colnames(q) <- c("spotted_bass", "northern_smallmouth_bass", "neosho_smallmouth_bass")


# get only genotyped samples from metadata
metadata <- metadata %>%
  filter(genotype == "yes")

# Bind genotyped sample_id column with population structure membership data
q <- cbind(metadata,
           q) %>%
  dplyr::select(sample_id, spotted_bass, northern_smallmouth_bass, neosho_smallmouth_bass)

# merge q data with phenotype data
phenotypes <- merge(q, 
                    phenotypes,
                    by = "sample_id")

```
## --------------  END OF ANALYSIS 5: GENETIC DIVERSITY AND DIFFERENTIATION ANALYSIS --------------  ##

