---
title: "NSMB_Microsats_MORPHOLOGY"
author: "Joe Gunn"
date: "7/25/2019"
output: html_document
---

## Libraries Needed For Analysis
```{r Libraries, echo = FALSE, include = FALSE}

library(riverdist)
library(sp)
library(readxl)
library(popgraph)
library(tidyverse)
library(gstudio)
library(ade4)
library(adegenet)
library(cowplot)
library(nlme)
library(poppr)
library(PopGenReport)
library(maps)
library(mapdata)
library(stringr)
library(rgdal)
library(sf)
library(ggsn)
library(raster)
library(lme4)
library(factoextra)
library(MASS)
library(logihist)
library(pophelper)
library(ggpubr)
library(DescTools)
library(vegan)
```

## Metadata
```{r}
smb_metadata_geo <- read_excel("../../data/metadata/smb_metadata_geo.xlsx")
smb_metadata_geo_nor <- smb_metadata_geo[c(617:766),]
smb_metadata_neo <- read_excel("../../data/metadata/smb_metadata_nonGEO.xlsx")
```

Data Info:

Here I provide summary information for my morphological analysis for easy access in the manuscript writing process. I give the number of samples with morphological data, and within those, the number of Northern SMB and Neosho SMB samples. I also give the number of individual samples that assigned 90 % or more to any one cluster and also had morphological data.

OMITTED AR24 BECAUSE IT ASSIGNS ALMOST 98 PERCENT TO SPOTTED BASS

############################################################################################################
SAMPLE SIZES

Total samples with morphological data = 249
Total Northern Range = 65
Total Neosho Range = 184

Total samples with both morphological data and genetic data = 185
Total samples assigning 90 % or more to neosho or northern = 88
Total Pure Neosho = 17
Total Pure Northern = 71
Total Admixed = 97

Total Length Data

Samples with Total Length data = 385
Neosho samples with Total Length data = 320
Northern samples with Total Length data = 65
Pure neosho Total Length samples = 55
Pure northern Total Length samples = 82
############################################################################################################

## Morphological Analysis 
## Read in data and clean
### Read in files, including the raw morphometric data for all samples, excluding any samples that have NA in the place of any morphometric measurements, as well as the proportion of the genotype assigning to each cluster within the K = 3 result, major clusters only (Neosho, Northern, or Spotted Bass)
```{r}
#All morphometric samples (even with those not used in genetic analysis because of poor amplification; see above for sample size)
morph_all <- read_excel("../../data/excel_data/all_morph_withpop.xlsx") #all morphological data for all samples
all_tl <- read_excel("../../data/excel_data/all_tl.xlsx") #only total length data for all samples (there are more of these individuals, because total length was taken on most samples, while all other measurements were only taken for a subset of samples)

#Neosho and Northern Range samples from full dataset
morph_all_neo <- morph_all %>% 
    filter(range == "Neosho")

morph_all_nor <- morph_all %>% 
    filter(range == "Northern")

#Total Length Samples 
tl_neo <- all_tl %>% 
    filter(range == "Neosho")

tl_nor <- all_tl %>% 
    filter(range == "Northern")
```

## Make data files for Pure Neosho and Northern based on K = 3
```{r}
all_samples_k3_proportions <- read_excel("../../data/excel_data/CERC_k3_proportions.xlsx") #Get genetic proportions from STRUCTURE for K = 3 on all samples

#Merge all morphometric samples with genetic samples (to include only those samples used in STRUCTURE analysis)
morph_gen_k3 <- merge(all_samples_k3_proportions, morph_all, by = "sample_id")
morph_gen_k3$rays <- as.integer(morph_gen_k3$rays) #number of fin rays should be an integer

#Determine which samples assign over 90% to a given cluster for ALL Data (not just total length)
morph_neo_90_k3 <- morph_gen_k3 %>% 
    subset(neosho_prop > 0.9)

morph_nor_90_k3 <- morph_gen_k3 %>% 
    subset(northern_prop > 0.9)

morph_neo_90_k3 <- as.data.frame(morph_neo_90_k3)
morph_nor_90_k3 <- as.data.frame(morph_nor_90_k3)

morph_neo_90_k3$gid <- as.factor(c(rep("Neosho", times = 17)))
morph_nor_90_k3$gid <- as.factor(c(rep("Northern", times = 71)))

morph_neo_90_k3 <- morph_neo_90_k3 [, -c(2:4)]
morph_nor_90_k3 <- morph_nor_90_k3 [, -c(2:4)]

colnames(morph_neo_90_k3) <- c("sample_id", "river", "range", "tl", "sl", "ol", "hl", "bd", "ray", "logtl", "logsl", "logol", "loghl", "logbd", "gid")
colnames(morph_nor_90_k3) <- c("sample_id", "river", "range", "tl", "sl", "ol", "hl", "bd", "ray", "logtl", "logsl", "logol", "loghl", "logbd", "gid")

morph_90_nor_neo_k3 <- rbind(morph_neo_90_k3, morph_nor_90_k3)
morph_90_all_k3 <- rbind(morph_neo_90_k3, morph_nor_90_k3)

#Admixture Analysis differenes (for preliminary analysis of 3rd chapter)

morph_no_northern <- morph_gen_k3 %>% 
    filter(!(range == "Northern")) #132 total rows

morph_pure_neo <- morph_no_northern %>%
  subset(neosho_prop > 0.9)

morph_admix_neo <- morph_no_northern %>%
  subset(neosho_prop < 0.9)

morph_pure_neo$gid <- factor(rep("pure", times = 17))
morph_admix_neo$gid <- factor(rep("admix", times = 115))

morph_pure_admix <- rbind(morph_pure_neo, morph_admix_neo)

ggplot(morph_pure_admix, aes(x = gid, y = logtl)) +
  geom_boxplot(show.legend = F, outlier.shape = NA, alpha = 0.6) +
  geom_point(position = position_jitter(width = 0.2), alpha = 0.6, color = "black", pch = 21, show.legend = F)

ggplot(morph_pure_admix, aes(x = logtl, y = logsl, fill = gid)) +
  geom_point(size =3 , alpha = 0.6, color = "black", pch = 21, show.legend = F) +
  geom_smooth(aes(color = gid), method = lm, se = F, show.legend = F) 

ggplot(morph_pure_admix, aes(x = logtl, y = logol, fill = gid)) +
  geom_point(size =3 , alpha = 0.6, color = "black", pch = 21, show.legend = F) +
  geom_smooth(aes(color = gid), method = lm, se = F, show.legend = F) 

ggplot(morph_pure_admix, aes(x = logtl, y = loghl, fill = gid)) +
  geom_point(size =3 , alpha = 0.6, color = "black", pch = 21, show.legend = F) +
  geom_smooth(aes(color = gid), method = lm, se = F, show.legend = F) 

ggplot(morph_pure_admix, aes(x = logtl, y = logbd, fill = gid)) +
  geom_point(size =3 , alpha = 0.6, color = "black", pch = 21, show.legend = F) +
  geom_smooth(aes(color = gid), method = lm, se = F, show.legend = F) 

```

## Summarize trait values - Calculate real (log-transformed) means for each morphometric trait
```{r}
#Means for morphometrics by RANGE
morph_all <- morph_all %>% 
    mutate(range = factor(range), river = factor(river))

sl_range_mean <- morph_all %>% 
    group_by(range) %>% 
    summarize(mean = mean(logsl))

ol_range_mean <- morph_all %>% 
    group_by(range) %>% 
    summarize(mean = mean(logol))

hl_range_mean <- morph_all %>%
    group_by(range) %>% 
    summarize(mean = mean(loghl))

bd_range_mean <- morph_all %>% 
    group_by(range) %>% 
    summarize(mean = mean(logbd))

morph_range_means <- morph_all %>% 
    group_by(range) %>% 
    summarize(mean_tl = mean(logtl), mean_sl = mean(logsl), mean_ol = mean(logol), mean_hl = mean(loghl), mean_bd = mean(logbd))


#Means for morphometrics by GENETIC ASSIGNMENT
morph_90_nor_neo_k3 <- morph_90_nor_neo_k3 %>% 
    mutate(gid = factor(gid), river = factor(river))

sl_gen_mean_k3 <- morph_90_nor_neo_k3 %>% 
    group_by(gid) %>% summarize(mean = mean(logsl))

ol_gen_mean_k3 <- morph_90_nor_neo_k3 %>% 
    group_by(gid) %>% summarize(mean = mean(logol))

hl_gen_mean_k3 <- morph_90_nor_neo_k3 %>% 
    group_by(gid) %>% summarize(mean = mean(loghl))

bd_gen_mean_k3 <- morph_90_nor_neo_k3 %>% 
    group_by(gid) %>% summarize(mean = mean(logbd)) 

morph_90_all_k3_means <- morph_90_all_k3 %>% 
    group_by(gid) %>% 
    summarize(mean_tl = mean(logtl), mean_sl = mean(logsl), mean_ol = mean(logol), mean_hl = mean(loghl), mean_bd = mean(logbd))
```

## Check for multicollinearity in total length. Is Total Length different between groups by RANGE or by GENETIC ASSIGNMENT?
```{r}
#Get the genetic data for just the total length data and clean for analysis
tl_gen_k3 <- merge(all_samples_k3_proportions, all_tl, by = "sample_id")

tl_neo_90_k3 <- tl_gen_k3 %>% 
    subset(neosho_prop > 0.9) #Get the individuals that assign 90 % to neosho SMB

tl_nor_90_k3 <- tl_gen_k3 %>% 
    subset(northern_prop > 0.9) #Get the individuals that assign 90 % to northern SMB

tl_neo_90_k3 <- as.data.frame(tl_neo_90_k3) #55 Samples from total length dataset that assign 90 % to neosho SMB
tl_nor_90_k3 <- as.data.frame(tl_nor_90_k3) #82 Samples from total length dataset that assign 90 % to northern SMB

tl_neo_90_k3$gid <- as.factor(c(rep("Neosho", times = 55)))
tl_nor_90_k3$gid <- as.factor(c(rep("Northern", times = 82)))
tl_90_neo_nor_k3 <- rbind(tl_neo_90_k3, tl_nor_90_k3)

tl_90_neo_nor_k3_merge <- merge(tl_90_neo_nor_k3, smb_metadata_geo, by = "sample_id")
all_tl_merge <- merge(all_tl, smb_metadata_geo, by = "sample_id")

lm_tl_gen_k3  <- (lme(logtl ~ gid, random = ~1|population, tl_90_neo_nor_k3_merge)) #NOT SIGNIFICANT
lm_tl_range <- (lme(logtl ~ range, random = ~1|population, all_tl_merge)) #NOT SIGNIFICANT

summary(lm_tl_gen_k3) #not significant (p = 0.109, F = 2.60)
summary(lm_tl_range) #not significant (p = 0.634, F = 0.23)

#Graph the collinearity
tl_range_diff <- ggplot(all_tl, aes(x = range, y = logtl)) +
    geom_point(position = position_jitter(width = 0.2), alpha = 0.4, size = 3) + 
    labs(x = "Range", y = "Log10 of Total Length (mm)", Title = "Range of Collection") + 
    theme_set(theme_cowplot(12)) +
    stat_summary(fun.y = mean, geom = "point", color = "blue", size = 4) + 
    stat_summary(fun.data = mean_sd, geom = "errorbar", width = 0.2, color = "blue", size = 2) + 
    theme(axis.title = element_text(size = 20)) + 
    theme(axis.text = element_text(size = 20))

tl_gen_diff_k3  <- ggplot(tl_90_neo_nor_k3, aes(x = gid, y = logtl)) +
    geom_point(position = position_jitter(width = 0.2), alpha = 0.4, size = 3) + 
    labs(x = "Assignment", Title = "Genetic Assignment") +
    theme_set(theme_cowplot(12)) +
    stat_summary(fun.y = mean, geom = "point", color = "blue", size = 4) + 
    stat_summary(fun.data = mean_sd, geom = "errorbar", width = 0.2, color = "blue", size = 2) + 
    theme(axis.title.y = element_blank()) + 
    theme(axis.title = element_text(size = 20)) + 
    theme(axis.text = element_text(size = 20))

#Plot Test for Collinearity Assumption 
pdf("/Users/joegunn/Desktop/Grad_School_Stuff/Research/Projects/Smallmouth_Bass_Microsatellites/visualization/morphology_figures/tl_collinearity_plots_k3.pdf", width = 13, height = 8)

plot_grid(tl_range_diff, tl_gen_diff_k3, nrow = 1, labels = c("A", "B"), label_size = 25)

dev.off()
```

## Check for deviations from normality in the range and genetic data to see if it is appropriate for linear modeling
```{r}
#Check for Normality in subspecies RANGES
#Shapiro-Wilk - if the p-value is above 0.05, it does not vary from normality

#Neosho Range
tl_neo_norm_test_range <- shapiro.test(morph_all_neo$logtl) #normal
sl_neo_norm_test_range <- shapiro.test(morph_all_neo$logsl) #normal
ol_neo_norm_test_range <- shapiro.test(morph_all_neo$logol) #not normal
hl_neo_norm_test_range <- shapiro.test(morph_all_neo$loghl) #normal
bd_neo_norm_test_range <- shapiro.test(morph_all_neo$logbd) #normal

#Northern Range
tl_nor_norm_test_range <- shapiro.test(morph_all_nor$logtl) #normal
sl_nor_norm_test_range <- shapiro.test(morph_all_nor$logsl) #normal
ol_nor_norm_test_range <- shapiro.test(morph_all_nor$logol) #normal
hl_nor_norm_test_range <- shapiro.test(morph_all_nor$loghl) #not normal
bd_nor_norm_test_range <- shapiro.test(morph_all_nor$logbd) #normal

#Neosho Genetics
tl_neo_norm_test_gen <- shapiro.test(morph_neo_90_k3$logtl) #normal
sl_neo_norm_test_gen <- shapiro.test(morph_neo_90_k3$logsl) #normal
ol_neo_norm_test_gen <- shapiro.test(morph_neo_90_k3$logol) #normal
hl_neo_norm_test_gen <- shapiro.test(morph_neo_90_k3$loghl) #normal
bd_neo_norm_test_gen <- shapiro.test(morph_neo_90_k3$logbd) #normal

#Northern Genetics
tl_nor_norm_test_gen <- shapiro.test(morph_nor_90_k3$logtl) #normal
sl_nor_norm_test_gen <- shapiro.test(morph_nor_90_k3$logsl) #normal
ol_nor_norm_test_gen <- shapiro.test(morph_nor_90_k3$logol) #normal
hl_nor_norm_test_gen <- shapiro.test(morph_nor_90_k3$loghl) #not normal
bd_nor_norm_test_gen <- shapiro.test(morph_nor_90_k3$logbd) #normal

```

# #QQ-Plots for Normality Test
```{r}
#Normality within Neosho
tl_neo_range_norm <- ggqqplot(morph_all_neo$logtl) + 
    labs(y = "Total Length Sample", title = "Neosho Range") + 
    theme_set(theme_cowplot(12)) + theme(axis.title.x = element_blank()) + 
    theme(plot.title = element_text(hjust = 0.5)) + 
    theme(axis.title.y = element_text(size = 20)) + 
    theme(plot.title = element_text(size = 20)) #good

sl_neo_range_norm <- ggqqplot(morph_all_neo$logsl) + 
    labs(y = "Standard Length Sample") + 
    theme_set(theme_cowplot(12)) + 
    theme(axis.title.x = element_blank()) + 
    theme(axis.title.y = element_text(size = 20)) #good

ol_neo_range_norm <- ggqqplot(morph_all_neo$logol) + 
    labs(y = "Orbital Length Sample") + 
    theme_set(theme_cowplot(12)) + 
    theme(axis.title.x = element_blank()) + 
    theme(axis.title.y = element_text(size = 20))#one weird sample

hl_neo_range_norm <- ggqqplot(morph_all_neo$loghl) + 
    labs(y = "Head Length Sample") + 
    theme_set(theme_cowplot(12)) + 
    theme(axis.title.x = element_blank()) + 
    theme(axis.title.y = element_text(size = 20)) #good

bd_neo_range_norm <- ggqqplot(morph_all_neo$logbd) + 
    labs(y = "Body Depth Sample", x = "Theoretical") + 
    theme_set(theme_cowplot(12)) + 
    theme(axis.title.y = element_text(size = 20)) + 
    theme(axis.title.x = element_text(size = 20)) #good
    
#Normality within Northern
tl_nor_range_norm <- ggqqplot(morph_all_nor$logtl) + 
    labs(title = "Northern Range") + 
    theme_set(theme_cowplot(12)) + 
    theme(axis.title.x = element_blank()) + 
    theme(axis.title.y = element_blank()) + 
    theme(plot.title = element_text(hjust = 0.5)) + 
    theme(plot.title = element_text(size = 20)) #good

sl_nor_range_norm <- ggqqplot(morph_all_nor$logsl) + 
    theme_set(theme_cowplot(12)) + 
    theme(axis.title.x = element_blank()) +
    theme(axis.title.y = element_blank()) #good

ol_nor_range_norm <- ggqqplot(morph_all_nor$logol) + 
    theme_set(theme_cowplot(12)) + 
    theme(axis.title.x = element_blank()) + 
    theme(axis.title.y = element_blank()) #good

hl_nor_range_norm <- ggqqplot(morph_all_nor$loghl) + 
    theme_set(theme_cowplot(12)) + 
    theme(axis.title.x = element_blank()) + 
    theme(axis.title.y = element_blank()) #4 weird samples

bd_nor_range_norm <- ggqqplot(morph_all_nor$logbd) + 
    labs(x = "Theoretical") + 
    theme_set(theme_cowplot(12)) + 
    theme(axis.title.x = element_text(size = 20)) + 
    theme(axis.title.y = element_blank())  #good
    
#Check for Normality in subspecies GENETIC ASSIGNMENTS
    
    #Within Neosho
tl_neo_genetics_norm <- ggqqplot(morph_neo_90_k3$logtl) + 
    labs(title = "Neosho Genetic Assignment") + 
    theme_set(theme_cowplot(12)) + 
    theme(axis.title.x = element_blank()) + 
    theme(axis.title.y = element_blank()) + 
    theme(plot.title = element_text(hjust = 0.5)) + 
    theme(plot.title = element_text(size = 20)) #good

sl_neo_genetics_norm <- ggqqplot(morph_neo_90_k3$logsl) + 
    theme_set(theme_cowplot(12)) +
    theme(axis.title.x = element_blank()) + 
    theme(axis.title.y = element_blank())  #good

ol_neo_genetics_norm <- ggqqplot(morph_neo_90_k3$logol) + 
    theme_set(theme_cowplot(12)) + 
    theme(axis.title.x = element_blank()) + 
    theme(axis.title.y = element_blank())  #good

hl_neo_genetics_norm <- ggqqplot(morph_neo_90_k3$loghl) + 
    theme_set(theme_cowplot(12)) + 
    theme(axis.title.x = element_blank()) + 
    theme(axis.title.y = element_blank())  #good

bd_neo_genetics_norm <- ggqqplot(morph_neo_90_k3$logbd) + 
    labs(x = "Theoretical") + theme_set(theme_cowplot(12)) +
    theme(axis.title.y = element_blank()) +
    theme(axis.title.x = element_text(size = 20)) #good
    
#Within Northern
tl_nor_genetics_norm <- ggqqplot(morph_nor_90_k3$logtl) + 
    labs(title = "Northern Genetic Assignment") + 
    theme_set(theme_cowplot(12)) + 
    theme(axis.title.x = element_blank()) + 
    theme(axis.title.y = element_blank()) + 
    theme(plot.title = element_text(hjust = 0.5)) + 
    theme(plot.title = element_text(size = 20)) #good

sl_nor_genetics_norm <- ggqqplot(morph_nor_90_k3$logsl) + 
    theme_set(theme_cowplot(12)) + 
    theme(axis.title.x = element_blank()) + 
    theme(axis.title.y = element_blank())  #good

ol_nor_genetics_norm <- ggqqplot(morph_nor_90_k3$logol) +
    theme_set(theme_cowplot(12)) + 
    theme(axis.title.x = element_blank()) +
    theme(axis.title.y = element_blank())  #good

hl_nor_genetics_norm <- ggqqplot(morph_nor_90_k3$loghl) + 
    theme_set(theme_cowplot(12)) + 
    theme(axis.title.x = element_blank()) + 
    theme(axis.title.y = element_blank())  #two weird samples

bd_nor_genetics_norm <- ggqqplot(morph_nor_90_k3$logbd) + 
    theme_set(theme_cowplot(12)) + 
    labs(x = "Theoretical") + 
    theme(axis.title.y = element_blank()) + 
    theme(axis.title.x = element_text(size = 20)) #one weird sample
    


#Put all of them together
neo_range_norm <- plot_grid(tl_neo_range_norm, sl_neo_range_norm, ol_neo_range_norm, hl_neo_range_norm, bd_neo_range_norm, nrow = 5, labels = c("p = 0.079", "p = 0.090", "p < 0.001 ", "p = 0.165", "p = 0.599"), label_x = 0.5, label_y = 0.35, label_size = 20)

nor_range_norm <- plot_grid(tl_nor_range_norm, sl_nor_range_norm, ol_nor_range_norm, hl_nor_range_norm, bd_nor_range_norm, nrow = 5, labels = c("p = 0.458", "p = 0.297", "p = 0.519 ", "p < 0.001", "p = 0.297"), label_x = 0.5, label_y = 0.35, label_size = 20)

neo_gen_norm <- plot_grid(tl_neo_genetics_norm, sl_neo_genetics_norm, ol_neo_genetics_norm, hl_neo_genetics_norm, bd_neo_genetics_norm, nrow = 5, labels = c("p = 0.446", "p = 0.521", "p = 0.509 ", "p = 0.685", "p = 0.203"), label_x = 0.5, label_y = 0.35, label_size = 20)

nor_gen_norm <- plot_grid(tl_nor_genetics_norm, sl_nor_genetics_norm, ol_nor_genetics_norm, hl_nor_genetics_norm, bd_nor_genetics_norm, nrow = 5, labels = c("p = 0.598", "p = 0.783", "p = 0.468", "p < 0.001 ", "p = 0.642"), label_x = 0.5, label_y = 0.35, label_size = 20)

#Combine all qq-plots into one supplementary figure
pdf("/Users/joegunn/Desktop/Grad_School_Stuff/Research/Projects/Smallmouth_Bass_Microsatellites/visualization/morphology_figures/qq_plots.pdf", width=17, height=17)

plot_grid(neo_range_norm, nor_range_norm, neo_gen_norm, nor_gen_norm, ncol = 4)

dev.off()
```

## density distributions for all traits
```{r}
#Probability density distributions for Samples by Range Assignment
tl_density <- ggplot(morph_even, aes(logtl, fill = range)) + geom_density()
sl_density <- ggplot(morph_even, aes(logsl, fill = range)) + geom_density() 
ol_density <- ggplot(morph_even, aes(logol, fill = range)) + geom_density()
hl_density <- ggplot(morph_even, aes(loghl, fill = range)) + geom_density()
bd_density <- ggplot(morph_even, aes(logbd, fill = range)) + geom_density()

#Probability density distributions for Samples by Genetic Assignment
tl_density_gen <- ggplot(morph_90_nor_neo_k3, aes(logtl, fill = gid)) + geom_density()
sl_density_gen <- ggplot(morph_90_nor_neo_k3, aes(logsl, fill = gid)) + geom_density() 
ol_density_gen <- ggplot(morph_90_nor_neo_k3, aes(logol, fill = gid)) + geom_density()
hl_density_gen <- ggplot(morph_90_nor_neo_k3, aes(loghl, fill = gid)) + geom_density()
bd_density_gen <- ggplot(morph_90_nor_neo_k3, aes(logbd, fill = gid)) + geom_density()

```

## Graph Distributions
```{r}
#Distributions labeled by RANGE
tl_dist_range <- ggplot(morph_all, aes(x = logtl, fill = range)) + 
    geom_density(show.legend = T) + 
    theme_set(theme_cowplot(12)) + 
    scale_fill_manual(values = c("springgreen3", "mediumpurple")) + 
    labs(fill = "Range", y = "Probability Density") + 
    theme(legend.position = c(0.6,0.9), legend.background = element_blank()) + 
    theme(legend.text = element_text(size = 15), legend.title = element_text(size = 20)) + 
    theme(axis.title.x = element_blank()) + 
    theme(axis.title.y = element_text(size = 20)) + 
    theme(axis.text = element_text(size = 20))

sl_dist_range <- ggplot(morph_all, aes(x = logsl, fill = range)) + 
    geom_density( show.legend = F) + 
    theme_set(theme_cowplot(12)) + 
    scale_fill_manual(values = c("springgreen3", "mediumpurple")) + 
    theme(axis.title.y = element_blank()) + 
    theme(axis.title.x = element_blank()) + 
    theme(axis.text = element_text(size = 20))

ol_dist_range <- ggplot(morph_all, aes(x = logol, fill = range)) + 
    geom_density( show.legend = F) + 
    theme_set(theme_cowplot(12)) + 
    scale_fill_manual(values = c("springgreen3", "mediumpurple")) + 
    theme(axis.title.y = element_blank()) + 
    theme(axis.title.x = element_blank()) + 
    theme(axis.text = element_text(size = 20)) + 
    theme(axis.text = element_text(size = 20))

hl_dist_range <- ggplot(morph_all, aes(x = loghl, fill = range)) + 
    geom_density( show.legend = F) + 
    theme_set(theme_cowplot(12)) + 
    scale_fill_manual(values = c("springgreen3", "mediumpurple")) + 
    theme(axis.title.y = element_blank()) + 
    theme(axis.title.x = element_blank()) + 
    theme(axis.text = element_text(size = 20))

bd_dist_range <- ggplot(morph_all, aes(x = logbd, fill = range)) + 
    geom_density( show.legend = F) + 
    theme_set(theme_cowplot(12)) + 
    scale_fill_manual(values = c("springgreen3", "mediumpurple")) + 
    theme(axis.title.x = element_blank()) + theme(axis.title.y = element_blank()) + 
    theme(axis.text = element_text(size = 20))

range_dist <- plot_grid(tl_dist_range, sl_dist_range, ol_dist_range, hl_dist_range, bd_dist_range, nrow = 1, labels = c("  a", "b", "c", "d", "e"), label_x = 0.1, label_y = 0.95, label_size = 25)


#Distributions labeled by GENETIC ASSIGNMENT
levels(morph_90_nor_neo_k3$gid) <- c("Neosho", "Northern")

tl_dist_gen <- ggplot(morph_90_nor_neo_k3, aes(x = logtl, fill = gid)) + 
    geom_density( show.legend = T) + 
    theme_set(theme_cowplot(12)) + 
    scale_fill_manual(values = c("springgreen3", "mediumpurple")) + 
    labs(fill = "Assignment", y = "Probability Density", x = "Total Length (mm)") + 
    theme(legend.position = c(0.6,0.9), legend.background = element_blank()) + 
    theme(legend.text = element_text(size = 15), legend.title = element_text(size = 20)) + 
    theme(axis.title = element_text(size = 20)) + 
    theme(axis.text = element_text(size = 20))

sl_dist_gen <- ggplot(morph_90_nor_neo_k3, aes(x = logsl, fill = gid)) + 
    geom_density( show.legend = F) + 
    theme_set(theme_cowplot(12)) + 
    scale_fill_manual(values = c("springgreen3", "mediumpurple")) + 
    labs(x = "Standard Length (mm)") + 
    theme(axis.title.y = element_blank()) + 
    theme(axis.title.x = element_text(size = 20)) + 
    theme(axis.text = element_text(size = 20))

ol_dist_gen <- ggplot(morph_90_nor_neo_k3, aes(x = logol, fill = gid)) + 
    geom_density( show.legend = F) + 
    theme_set(theme_cowplot(12)) + 
    scale_fill_manual(values = c("springgreen3", "mediumpurple")) + 
    labs(x = "Orbital Length (mm)") + 
    theme(axis.title.y = element_blank()) + 
    theme(axis.title.x = element_text(size = 20)) + 
    theme(axis.text = element_text(size = 20))

hl_dist_gen <- ggplot(morph_90_nor_neo_k3, aes(x = loghl, fill = gid)) + 
    geom_density( show.legend = F) + 
    theme_set(theme_cowplot(12)) + 
    scale_fill_manual(values = c("springgreen3", "mediumpurple")) + 
    labs(x = "Head Length (mm)") + 
    theme(axis.title.y = element_blank()) + 
    theme(axis.title.x = element_text(size = 20)) + 
    theme(axis.text = element_text(size = 20))

bd_dist_gen <- ggplot(morph_90_nor_neo_k3, aes(x = logbd, fill = gid)) + 
    geom_density( show.legend = F) + 
    theme_set(theme_cowplot(12)) + 
    scale_fill_manual(values = c("springgreen3", "mediumpurple")) + 
    labs(x = "Body Depth (mm)") + 
    theme(axis.title.y = element_blank()) + 
    theme(axis.title.x = element_text(size = 20)) + 
    theme(axis.text = element_text(size = 20))

gen_dist <- plot_grid(tl_dist_gen, sl_dist_gen, ol_dist_gen, hl_dist_gen, bd_dist_gen, nrow = 1, labels = c("   f", "g", "h", "i", "j"), label_x = 0.1, label_y = 0.95, label_size = 25)

###Plot everything together
pdf("/Users/joegunn/Desktop/Grad_School_Stuff/Research/Projects/Smallmouth_Bass_Microsatellites/visualization/morphology_figures/morph_distributions.pdf", width=20, height=8)

plot_grid(range_dist, gen_dist, nrow = 2)

dev.off()
```

Morphometric ANCOVA by RANGE and GENETIC ASSIGNMENT

Information on method:

The iSMSC Method from Berner 2010 suggests that each individual morphometric trait be independently regressed onto a variable of body size (in my case, total length) in order to partial out the effect of that size trait. Once each trait has been analyzed in a general linear model (ANCOVA) against total length, you can find the grand mean of total length across all pooled samples (including samples from every group analyzed - ie. ranges), and you can calculate the average value of each morphometric trait at THAT grand mean value of total length. This will allow direct comparison between populations (using the unique slope and intercept coefficients for each population), without the biasing effect of total length

## By Range
## Univariate ANCOVA with Total Length as the covariate
```{r}
###Samples by Range###
#ANCOVA interaction models
ancova_sl_range_interaction <- lme(logsl ~ logtl*range, random = ~1|river, data = morph_all)
ancova_ol_range_interaction <- lme(logol ~ logtl*range, random = ~1|river, data = morph_all)
ancova_hl_range_interaction <- lme(loghl ~ logtl*range, random = ~1|river, data = morph_all)
ancova_bd_range_interaction <- lme(logbd ~ logtl*range, random = ~1|river, data = morph_all)

#Stat summaries
anova(ancova_sl_range_interaction) #no interaction (p = 0.271), F = 1.200
anova(ancova_ol_range_interaction) #interaction (p < 0.001), F = 30.628
anova(ancova_hl_range_interaction) #no interaction (p = 0.636), F = 0.220
anova(ancova_bd_range_interaction) #no interaction (p = 0.289), F = 1.13

#ANCOVA additive models
ancova_sl_range_additive <- lme(logsl ~ logtl + range, random = ~1|river, data = morph_all)
ancova_ol_range_additive <- lme(logol ~ logtl + range, random = ~1|river, data = morph_all)
ancova_hl_range_additive <- lme(loghl ~ logtl + range, random = ~1|river, data = morph_all) 
ancova_bd_range_additive <- lme(logbd ~ logtl + range, random = ~1|river, data = morph_all)

#Stat summaries
anova(ancova_sl_range_additive) #not significant (p = 0.166), F = 2.100
anova(ancova_ol_range_additive) #not significant (p = 0.7443), F = 0.110
anova(ancova_hl_range_additive) #significant (p = 0.001), F = 14.070
anova(ancova_bd_range_additive) #not significant (p = 0.696), F = 0.160
```

## Get coefficients from ANCOVAS
```{r}
#slope and intercept information for each range
coefficients_sl_range <- lmList(logsl ~ logtl|range, data = morph_all)
coefficients_ol_range <- lmList(logol ~ logtl|range, data = morph_all)
coefficients_hl_range <- lmList(loghl ~ logtl|range, data = morph_all)
coefficients_bd_range <- lmList(logbd ~ logtl|range, data = morph_all)

tl_grandmean_all <- morph_all %>% summarize(g_mean = mean(logtl)) #tl grandmean = 2.452

summary(coefficients_sl_range)
    #Neosho intercept = -0.103
    #Neosho slope = 1.007
    #Northern intercept = -0.019
    #Northern slope = 0.977
    #t value (range) = 1.143
    #t value (interaction) = -1.013
    #P (range, additive model) = <0.001
    #P (interaction) = 0.312
    #R2 = 0.961
    #Residual standard error = 0.015
    #Calculated Neosho Average = 
    #Calculated Northern Average = 

summary(coefficients_ol_range)
    #Neosho intercept = -0.264
    #Neosho slope = 0.576
    #Northern intercept = 1.237
    #Northern slope = -0.033
    #t value (range) = 5.240
    #t value (interaction) = -.5228
    #P (interaction) = < 0.001
    #R2 = 0.284
    #Residual standard error = 0.059
    #Calculated Neosho Average = 
    #Calculated Northern Average = 

summary(coefficients_hl_range)
    #Neosho intercept = -0.426
    #Neosho slope = 0.949
    #Northern intercept = -0.523
    #Northern slope = 0.974
    #t value (range) = -0.550
    #t value (interaction) = 0.353
    #P (range, additive model) = <0.001
    #P (interaction) = 0.724
    #R2 = 0.799
    #Residual standard error = 0.036
    #Calculated Neosho Average = 
    #Calculated Northern Average = 

summary(coefficients_bd_range)
    #Neosho intercept = -0.950
    #Neosho slope = 1.129
    #Northern intercept = -1.299
    #Northern slope = 1.269
    #t value (range) = -1.896
    #t value (interaction) = 1.862
    #P (range, additive model) = 0.224
    #P (interaction) = 0.064
    #R2 = 0.841
    #Residual standard error = 0.038
    #Calculated Neosho Average = 
    #Calculated Northern Average = 
```

## By Genetic Assignment
## ANCOVAs with total length as a covariate
```{r}
#ANCOVA interaction models by Genetic Assignment
ancova_sl_gen_interaction <- lme(logsl ~ logtl*gid, random = ~1|river, data = morph_90_nor_neo_k3)
ancova_ol_gen_interaction <- lme(logol ~ logtl*gid, random = ~1|river, data = morph_90_nor_neo_k3)
ancova_hl_gen_interaction <- lme(loghl ~ logtl*gid, random = ~1|river, data = morph_90_nor_neo_k3)
ancova_bd_gen_interaction <- lme(logbd ~ logtl*gid, random = ~1|river, data = morph_90_nor_neo_k3)

#Stat summaries
anova(ancova_sl_gen_interaction) #no interaction (p = 0.181), F = 1.800
anova(ancova_ol_gen_interaction) #no interaction (p = 0.158), F = 2.041
anova(ancova_hl_gen_interaction) #no interaction (p = 0.728), F = 0.120
anova(ancova_bd_gen_interaction) #no interaction (p = 0.693), F = 0.160

#ANCOVA additive models
ancova_sl_gen_additive <- lme(logsl ~ logtl + gid, random = ~1|river, data = morph_90_nor_neo_k3)
ancova_ol_gen_additive <- lme(logol ~ logtl + gid, random = ~1|river, data = morph_90_nor_neo_k3)
ancova_hl_gen_additive <- lme(loghl ~ logtl + gid, random = ~1|river, data = morph_90_nor_neo_k3)
ancova_bd_gen_additive <- lme(logbd ~ logtl + gid, random = ~1|river, data = morph_90_nor_neo_k3)

#Stat summaries
anova(ancova_sl_gen_additive) #not significant (p = 0.620), F = 0.300
anova(ancova_ol_gen_additive) #not significant (p = 0.756), F = 0.100
anova(ancova_hl_gen_additive) #not significant (p = 0.097), F = 3.140
anova(ancova_bd_gen_additive) #not significant (p = 0.681), F = 0.180
```

```{r}
#slope and intercept information for each range
coefficients_sl_gen <- lmList(logsl ~ logtl|gid, data = morph_90_nor_neo_k3)
coefficients_ol_gen <- lmList(logol ~ logtl|gid, data = morph_90_nor_neo_k3)
coefficients_hl_gen <- lmList(loghl ~ logtl|gid, data = morph_90_nor_neo_k3)
coefficients_bd_gen <- lmList(logbd ~ logtl|gid, data = morph_90_nor_neo_k3)

tl_grandmean_gen <- morph_90_nor_neo_k3 %>% summarize(g_mean = mean(logtl)) #tl grandmean = 2.447

summary(coefficients_sl_gen)
    #Neosho intercept = -0.435
    #Neosho slope = 1.145
    #Northern intercept = -0.066
    #Northern slope = 0.996
    #t value (range) = 2.306
    #t value (interaction) = -2.248
    #P (range) = 0.024
    #P (interaction) = 0.027
    #R2 = 0.963
    #Residual standard error = 0.014
    #Calculated Neosho Average = 
    #Calculated Northern Average = 

summary(coefficients_ol_gen)
    #Neosho intercept = 0.056
    #Neosho slope = 0.452
    #Northern intercept = 0.862
    #Northern slope = 0.122
    #t value (range) = 1.092
    #t value (interaction) = -1.076
    #P (range) = 0.278
    #P (interaction) = 0.285
    #R2 = 0.026
    #Residual standard error = 0.066
    #Calculated Neosho Average = 
    #Calculated Northern Average = 

summary(coefficients_hl_gen)
    #Neosho intercept = -0.412 
    #Neosho slope = 0.945
    #Northern intercept = -0.620
    #Northern slope = 1.019
    #t value (range) = -0.355
    #t value (interaction) = 0.303
    #P (range) = 0.723
    #P (interaction) = 0.763
    #R2 = 0.624
    #Residual standard error = 0.053
    #Calculated Neosho Average = 
    #Calculated Northern Average = 

summary(coefficients_bd_gen)
    #Neosho intercept = -1.618
    #Neosho slope = 1.407
    #Northern intercept = -1.177
    #Northern slope = 1.219
    #t value (range) = 0.921
    #t value (interaction) = -0.942
    #P (range) = 0.359
    #P (interaction) = 0.349
    #R2 = 0.804
    #Residual standard error = 0.043
    #Calculated Neosho Average = 
    #Calculated Northern Average = 

```

## Plot ANCOVA
```{r}

#All morphological samples 
sl_tl_range <- ggplot() + 
    geom_point(data = morph_all_neo, aes(x = logtl, y = logsl, group = range, fill = range, shape = range), size = 3.5, pch = 21, color = "black", show.legend = T) +
    geom_smooth(data = morph_all_neo, aes(x = logtl, y = logsl, color = range), show.legend = F, method = "lm") +
    geom_point(data = morph_all_nor, aes(x = logtl, y = logsl, fill = range, shape = range), size = 3.5, pch = 22, color = "black", show.legend = F) +
    geom_smooth(data = morph_all_nor, aes(x = logtl, y = logsl, color = range), show.legend = F, method = "lm") +
    theme_set(theme_cowplot(12)) + 
    scale_fill_manual(values = c("springgreen3", "mediumpurple")) +
    scale_color_manual(values = c("springgreen3", "mediumpurple")) +
    scale_shape_manual(values = c("circle","square")) +
    labs(y = "Standard Length (mm)", fill = "Range", color = "Range", shape = "Range", title = "Taxonomic Range") +
    theme(plot.title = element_text(hjust = 0.5, face = "bold", size = 20)) +
    theme(axis.title.x = element_blank()) + 
    theme(axis.text = element_text(size = 20)) + 
    theme(axis.title.y = element_text(size = 20)) +
    theme(legend.position = c(0.7,0.2)) +
    theme(legend.title = element_text(size = 20, face = "bold")) +
    theme(legend.background = element_blank()) + 
    theme(legend.text = element_text(size = 20))

ol_tl_range <- ggplot() + 
    geom_point(data = morph_all_neo, aes(x = logtl, y = logol, group = range, fill = range, shape = range), size = 3.5, pch = 21, color = "black", show.legend = F) +
    geom_smooth(data = morph_all_neo, aes(x = logtl, y = logol, color = range), show.legend = F, method = "lm") +
    geom_point(data = morph_all_nor, aes(x = logtl, y = logol, fill = range, shape = range), size = 3.5, pch = 22, color = "black", show.legend = F) +
    geom_smooth(data = morph_all_nor, aes(x = logtl, y = logol, color = range), show.legend = F, method = "lm") +
    theme_set(theme_cowplot(12)) + 
    scale_fill_manual(values = c("springgreen3", "mediumpurple")) +
    scale_color_manual(values = c("springgreen3", "mediumpurple")) +
    scale_shape_manual(values = c("circle","square")) +
    labs(y = "Orbital Length (mm)", fill = "Range", color = "Range", shape = "Range") + 
    theme(axis.title.x = element_blank()) + 
    theme(axis.text = element_text(size = 20)) + 
    theme(axis.title.y = element_text(size = 20))

hl_tl_range <- ggplot() + 
    geom_point(data = morph_all_neo, aes(x = logtl, y = loghl, group = range, fill = range, shape = range), size = 3.5, pch = 21, color = "black", show.legend = F) +
    geom_smooth(data = morph_all_neo, aes(x = logtl, y = loghl, color = range), show.legend = F, method = "lm") +
    geom_point(data = morph_all_nor, aes(x = logtl, y = loghl, fill = range, shape = range), size = 3.5, pch = 22, color = "black", show.legend = F) +
    geom_smooth(data = morph_all_nor, aes(x = logtl, y = loghl, color = range), show.legend = F, method = "lm") +
    theme_set(theme_cowplot(12)) + 
    scale_fill_manual(values = c("springgreen3", "mediumpurple")) +
    scale_color_manual(values = c("springgreen3", "mediumpurple")) +
    scale_shape_manual(values = c("circle","square")) +
    labs(y = "Head Length (mm)", fill = "Range", color = "Range", shape = "Range") +
    theme(axis.title.x = element_blank()) + 
    theme(axis.text = element_text(size = 20)) + 
    theme(axis.title.y = element_text(size = 20))

bd_tl_range <- ggplot() + 
    geom_point(data = morph_all_neo, aes(x = logtl, y = logbd, group = range, fill = range, shape = range), size = 3.5, pch = 21, color = "black", show.legend = F) +
    geom_smooth(data = morph_all_neo, aes(x = logtl, y = logbd, color = range), show.legend = F, method = "lm") +
    geom_point(data = morph_all_nor, aes(x = logtl, y = logbd, fill = range, shape = range), size = 3.5, pch = 22, color = "black", show.legend = F) +
    geom_smooth(data = morph_all_nor, aes(x = logtl, y = logbd, color = range), show.legend = F, method = "lm") +
    theme_set(theme_cowplot(12)) + 
    scale_fill_manual(values = c("springgreen3", "mediumpurple")) +
    scale_color_manual(values = c("springgreen3", "mediumpurple")) +
    scale_shape_manual(values = c("circle","square")) +
    labs(y = "Body Depth (mm)", x = "Total Length (mm)", fill = "Range", color = "Range", shape = "Range") +
    theme(axis.text = element_text(size = 20)) + 
    theme(axis.title = element_text(size = 20))

range_grid <- plot_grid(sl_tl_range, ol_tl_range, hl_tl_range, bd_tl_range, nrow = 4, labels = c("a","c","e","g"), label_x = 0.15, label_y = 0.95, label_size = 25)


#Only Pure Samples
sl_tl_gen <- ggplot() + 
    geom_point(data = morph_neo_90_k3, aes(x = logtl, y = logsl, group = gid, fill = gid, shape = gid), size = 3.5, pch = 21, color = "black", show.legend = T) +
    geom_smooth(data = morph_neo_90_k3, aes(x = logtl, y = logsl, color = gid), show.legend = F, method = "lm") +
    geom_point(data = morph_nor_90_k3, aes(x = logtl, y = logsl, fill = gid, shape = gid), size = 3.5, pch = 22, color = "black", show.legend = F) +
    geom_smooth(data = morph_nor_90_k3, aes(x = logtl, y = logsl, color = gid), show.legend = F, method = "lm") +
    theme_set(theme_cowplot(12)) + 
    scale_fill_manual(values = c("springgreen3", "mediumpurple")) +
    scale_color_manual(values = c("springgreen3", "mediumpurple")) +
    scale_shape_manual(values = c("circle","square")) +
    labs(y = "Standard Length (mm)", fill = "Assignment", color = "Assignment", shape = "Assignment", title = "Genetic Assignment") +
    theme(plot.title = element_text(hjust = 0.5, face = "bold", size = 20)) +
    theme(axis.title.x = element_blank()) + 
    theme(axis.text = element_text(size = 20)) + 
    theme(axis.title.y = element_blank()) +
    theme(legend.position = c(0.7,0.2)) +
    theme(legend.title = element_text(size = 20, face = "bold")) +
    theme(legend.background = element_blank()) + 
    theme(legend.text = element_blank())

ol_tl_gen <- ggplot() + 
    geom_point(data = morph_neo_90_k3, aes(x = logtl, y = logol, group = gid, fill = gid, shape = gid), size = 3.5, pch = 21, color = "black", show.legend = F) +
    geom_smooth(data = morph_neo_90_k3, aes(x = logtl, y = logol, color = gid), show.legend = F, method = "lm") +
    geom_point(data = morph_nor_90_k3, aes(x = logtl, y = logol, fill = gid, shape = gid), size = 3.5, pch = 22, color = "black", show.legend = F) +
    geom_smooth(data = morph_nor_90_k3, aes(x = logtl, y = logol, color = gid), show.legend = F, method = "lm") +
    theme_set(theme_cowplot(12)) + 
    scale_fill_manual(values = c("springgreen3", "mediumpurple")) +
    scale_color_manual(values = c("springgreen3", "mediumpurple")) +
    scale_shape_manual(values = c("circle","square")) +
    labs(y = "Orbital Length (mm)", fill = "gid", color = "gid", shape = "gid") + 
    theme(axis.title.x = element_blank()) + 
    theme(axis.text = element_text(size = 20)) + 
    theme(axis.title.y = element_blank())

morph_nor_90_k3

hl_tl_gen <- ggplot() + 
    geom_point(data = morph_neo_90_k3, aes(x = logtl, y = loghl, group = gid, fill = gid, shape = gid), size = 3.5, pch = 21, color = "black", show.legend = F) +
    geom_smooth(data = morph_neo_90_k3, aes(x = logtl, y = loghl, color = gid), show.legend = F, method = "lm") +
    geom_point(data = morph_nor_90_k3, aes(x = logtl, y = loghl, fill = gid, shape = gid), size = 3.5, pch = 22, color = "black", show.legend = F) +
    geom_smooth(data = morph_nor_90_k3, aes(x = logtl, y = loghl, color = gid), show.legend = F, method = "lm") +
    theme_set(theme_cowplot(12)) + 
    scale_fill_manual(values = c("springgreen3", "mediumpurple")) +
    scale_color_manual(values = c("springgreen3", "mediumpurple")) +
    scale_shape_manual(values = c("circle","square")) +
    labs(y = "Head Length (mm)", fill = "gid", color = "gid", shape = "gid") +
    theme(axis.title.x = element_blank()) + 
    theme(axis.text = element_text(size = 20)) + 
    theme(axis.title.y = element_blank())

bd_tl_gen <- ggplot() + 
    geom_point(data = morph_neo_90_k3, aes(x = logtl, y = logbd, group = gid, fill = gid, shape = gid), size = 3.5, pch = 21, color = "black", show.legend = F) +
    geom_smooth(data = morph_neo_90_k3, aes(x = logtl, y = logbd, color = gid), show.legend = F, method = "lm") +
    geom_point(data = morph_nor_90_k3, aes(x = logtl, y = logbd, fill = gid, shape = gid), size = 3.5, pch = 22, color = "black", show.legend = F) +
    geom_smooth(data = morph_nor_90_k3, aes(x = logtl, y = logbd, color = gid), show.legend = F, method = "lm") +
    theme_set(theme_cowplot(12)) + 
    scale_fill_manual(values = c("springgreen3", "mediumpurple")) +
    scale_color_manual(values = c("springgreen3", "mediumpurple")) +
    scale_shape_manual(values = c("circle","square")) +
    labs(y = "Body Depth (mm)", x = "Total Length (mm)", fill = "gid", color = "gid", shape = "gid") +
    theme(axis.text = element_text(size = 20)) + 
    theme(axis.title.x = element_text(size = 20)) + 
    theme(axis.title.y = element_blank())

gen_grid <- plot_grid(sl_tl_gen, ol_tl_gen, hl_tl_gen, bd_tl_gen, nrow = 4, labels = c("b","d","f","h"), label_x = 0.1, label_y = 0.95, label_size = 25)


###Plot everything together
pdf("/Users/joegunn/Desktop/Grad_School_Stuff/Research/Projects/Smallmouth_Bass_Microsatellites/visualization/morphology_figures/morph_size_corrections.pdf", width=12, height=20)

plot_grid(range_grid, gen_grid, ncol = 2)

dev.off()


#extra pdf for presentation (TWS-AFS 2019 in Reno, NV)
pdf("/Users/joegunn/Desktop/Grad_School_Stuff/Research/Projects/Smallmouth_Bass_Microsatellites/visualization/morphology_figures/hl_graph_for_presentation.pdf", width=7, height=5)

ggplot(morph_all, aes(x = logtl, y = loghl, color = range)) + 
    theme_set(theme_cowplot(12)) + 
    geom_point(show.legend = F, size = 2) + 
    geom_smooth(show.legend = F, method = "lm") + 
    scale_color_manual(values = c("springgreen3", "mediumpurple")) + 
    labs(y = "Head Length (mm)", x = "Total Length (mm)") + 
    theme(axis.text = element_text(size = 20)) + 
    theme(axis.title = element_text(size = 20)) + 
    theme(plot.background = element_rect(fill = "black")) +
    theme(axis.line = element_line(color = "white")) + 
    theme(axis.ticks = element_line(color = "white")) +
    theme(axis.title = element_text(color = "white")) + 
    theme(axis.text = element_text(color = "white"))

dev.off()
```

## Raw Morphometric Analysis - Discriminant Function Analysis
```{r}
#DFA for Range Assignment
morph_all_dfa <- morph_all[,-c(1,4:8)]
morph_all_dfa <- as.data.frame(morph_all_dfa)
morph_all_dfa <- morph_all_dfa %>% drop_na()

#RunDFA
ldafm_range <- lda(range ~ logtl + logsl + logol + loghl + logbd, data = morph_all_dfa) 

#Get DFA predicted values
lda.range.p<-predict(ldafm_range, CV = TRUE, morph_all_dfa[c(4:8)])$class
table(lda.range.p, morph_all_dfa[,2])
tab <- table(lda.range.p, morph_all_dfa[,2])
summary(tab)
diag(prop.table(tab,1))
sum(diag(prop.table(tab)))

#DFA for Genetic Assignment
morph_90_dfa <- morph_90_nor_neo_k3[,-c(1:9)]
morph_90_dfa <- as.data.frame(morph_90_dfa)
morph_90_dfa <- morph_90_dfa %>% 
    drop_na()

ldafm_gen <- lda(gid ~ logtl + logsl + logol + loghl + logbd, data = morph_90_dfa)
ldafm_gen_cv <- lda(gid ~ logtl + logsl + logol + loghl + logbd, data = morph_90_dfa, CV = TRUE) 

lda.gen.p <- predict(ldafm_gen, CV = TRUE, morph_90_dfa[c(1:5)])$class
tab_gen <- table(lda.gen.p, morph_90_dfa[,6])
summary(tab_gen)
diag(prop.table(tab_gen,1))
sum(diag(prop.table(tab_gen)))
```

## Logistic Regression of Fin Ray Data
```{r}

#Do a logistic regression for fin ray count assessed by the proportion of the genotype assigning to the Neosho cluster. What is the probability that higher genomic proportion assigning to Neosho results in fin ray count closer to 13 (as we would expect?)

logitTransform <- function(x) { #function to produce logistic regression values
  log(x/(1-x))
}


morph_gen_logit <- morph_gen_k3 %>%
    mutate(logitNeo = logitTransform(neosho_prop))

morph_gen_logit$rays[morph_gen_logit$rays == "13"] <- "1"
morph_gen_logit$rays[morph_gen_logit$rays == "14"] <- "0"

morph_gen_logit <- morph_gen_logit %>% 
    mutate(rays = as.numeric(rays))

#Run logistic regression general linear model. This model will show the probability of having 13 or 14 soft dorsal fin rays based on the genetic assignment of individuals to either Northern or Neosho Smallmouth Bass. 

logistic_rays <- glm(rays ~ neosho_prop, data = morph_gen_logit, family = binomial(link = "logit"))
summary(logistic_rays)
PseudoR2(logistic_rays)

#plot logistic regression

pdf("/Users/joegunn/Desktop/Grad_School_Stuff/Research/Projects/Smallmouth_Bass_Microsatellites/visualization/morphology_figures/fin_logistic_regression.pdf", width=12, height=8)

ggplot(morph_gen_logit, aes(x = neosho_prop, y = rays)) + 
    geom_point(show.legend = F, size = 4) + 
    geom_smooth(method="glm", method.args = list(family = "binomial"), show.legend = TRUE, color = "red") + 
    labs(x = "Genotypic Proportion Assigned to the Neosho Cluster", y = "Probability of Fin Ray Count") + 
    theme(axis.text = element_text(size = 20)) +
    theme(axis.title = element_text(size = 20))

dev.off()
```
