---
title: "Analysis 3: Hierarchical population structure analysis"
author: "Joe Gunn"
date: "2022-07-28"
output: html_document
---

# Project: Population genetic structure and morphological differentiation between Northern Smallmouth Bass (<i>Micropterus dolomieu dolomieu</i>) and Neosho Smallmouth Bass (<i>M. d. velox</i>)
We investigated patterns of genetic (via microsatellites) and morphological diversity, differentiation, and structure across Smallmouth Bass populations (<i>Micropterus dolomieu</i>) in the Central Interior Highlands (CIH) of North America, including the two recognized subspecies: Northern Smallmouth Bass (<i>M. d. dolomieu</i>), which is native to the lower Ozark Highlands, and Neosho Smallmouth Bass (<i>M. d. velox</i>), which is endemic to tributaries of the Arkansas River Basin. We compared three independent combinations of starting parameters in the Bayesian clustering software program STRUCTURE to ascertain a robust picture of hierarchical genetic structure in the CIH. We paired these data with genetic diversity metrics (e.g., observed and expected heterozygosity and allelic richness) to determine relative amounts of variation across geographically separated populations. We then assessed differentiation in five morphometric and meristic traits. We ultimately aimed to validate or amend the taxonomic status of the Smallmouth Bass subspecies by revealing potential ecological and evolutionary divergence between them, with the hope that increased taxonomic resolution would provide insight into the presence and distribution of evolutionary significant and management units for a popular sportfish.

## Specific Aim: Hierarchical structure analysis with three independent methods
For this aim, we use Bayesian clustering analysis and genotypes from 14 microsatellite loci with STRUCTURE (see citation below in under "Programs Needed") to assess patterns hierarchical genetic structure among and within black bass species (i.e., Spotted Bass and Smallmouth Bass) and subspecies (i.e., Northern Smallmouth Bass and Neosho Smallmouth Bass) in the CIH. To maximize our ability to ascertain potentially complex or cryptic levels of genetic diversity, we employ and integrate three independent "methods" within STRUCTURE, which each use either different starting parameters for identifying genetic clusters or different inference techniques for determining the optimal number of clusters: 1) <i>default parameters</i> (default parameters in STRUCUTRE); 2) <i>Wang parameters</i> (parameters recommended by Wang (2017); and 3) <i>Puechmaille metrics</i> (optimal cluster inference method recommended by Puechmaille 2016).

## Phases of Analysis
### Phase 1: Genotype data preparation
### Phase 2: Structure analysis
### Phase 3: Principal Components Analysis

Programs needed:

STRUCTURE v.2.3.4. (Pritchard et al. 2000)

Citation:

Pritchard JK, Stephens M, Donnelly P. 2000. Inference of population structure using multilocus genotype data. Genetics 155: 945-959.

CLUMPP v.1.1.2 (Jakobsson and Rosenberg 2007)

Citation:

Jakobsson M, Rosenberg NA. 2007. CLUMPP: A cluster matching and permutation program for dealing with label switching and multimodality in analysis of population structure. Bioinformatics 23: 1801-1806.

We ran each hierarchical analysis in separate .Rmd files, which are found in the `code/` directory.

## Libraries needed for analysis
```{r}
library(tidyverse)
library(cowplot)
library(readxl)
library(writexl)
library(genepopedit)
library(adegenet)
library(gstudio)
```

## PHASE 1: GENOTYPE DATA PREPARATION
In this phase of the analysis, we convert the raw microsatellite genotype data (`../raw_data/genotype_data.xlsx`) into Genepop-compatible format, which is read by the software program STRUCTURE in all downstream analyses. We use the same exact genepop data for each independent analysis in STRUCTURE; thus, we generate a single genepop and subsequent structure file, which includes all samples representing all collection sites.

We also generate two additional genepop files to be used for substructure analysis using default STRUCTURE parameters. We generate one file with only Neosho Smallmouth Bass, and one file with only Northern Smallmouth Bass. This step follows from STRUCTURE results of Phase 2, Step 2 (default parameter model).

### STEP 1: Manually reformat Excel genotype data into genepop-compatible format.
In this step, we opted to reformat the raw Excel microsatellite genotype data by hand in Excel, given that we could not find a working R library and/or function that converts raw genotype tables into genepop format (i.e., we attempted to use the <i>genind_to_genepop()</i> function from the package <i>graph4lg</i>, but we could not get our genind to convert properly). We followed the steps below exactly in Excel to convert our genotype data into Genepop format, and we saved the corresponding data as a text file in the working `data` directory.

#### 1a. Add a "," after each sample_id in the first column.

#### 1b. Insert a row above each new population in the data, and insert "pop" in the sample_id column in each empty row (the word "pop" should appear for each population)

##### 1c. Copy the header row of microsatellite locus IDs and transpose paste in a new Excel sheet; each locus ID should be listed in its own row (number of rows should be equal to numbrer of locus IDs)

##### 1d. Copy all genotype and label data from the original Excel file and paste in the row immediately below the list of locus IDs

##### 1e. Save the Excel sheet as a .xlsx and a .txt file, with a specification that it is a "genepop" file.

### STEP 2: Generate STRUCTURE formatted file from genepop format.
In this step, we read in, check, and convert the genepop file manually generated in STEP 1 above into a STRUCTURE formatted file. There are quite a few tricky and very specific formatting needs for the genepop text file before genepopedit will convert successfully.

Follow Step 2a-2f closely to create a proper genepop format. Click on the text file manually here, in R Studio, and use Ctrl + F in R Studio to find and replace the necessary things.

In this step, we are reading in, checking, and converting the genepop file manually generated in STEP 1 above into a STRUCTURE formatted file. There are quite a few tricky and very specific formatting needs for the genepop text file before genepopedit will convert successfully. Follow Step 2a-2f closely to create a proper genepop format. Click on the text file manually here, in R Studio, and use Ctrl + F in R Studio to find and replace the necessary things.

#### 2a. All quotation marks were removed from the text file

#### 2b. All "," in the genepop text files were manually replaced with " ,  "

#### 2c. All "tabs" between six-digit alleles were were manually replaced with " " (a single space)

#### 2d. The top row designating the dataset combination was manually omitted

#### 2e. Any additional space at the bottom of the text file was removed

#### 2f. Generate STRUCTURE file from genepop file; run the Rmd chunk below.
Here it is important to not include the 'popgroup = ' unless it is explicitly necessary. Assigning popgroup to a dataframe with population names will cause the STRUCTURE input file to have word strings in the popdata column. Our experience is that STRUCTURE expects an integer here rather than a string, so best not to use popgroup.

##### Convert genepop to STRUCTURE format:`
```{r}
# Generate structure .txt file from genepop .txt file
genepop_structure("data/genepop_data/genepop.txt", 
                  locusnames = TRUE, 
                  path = "data/structure_data/input_data/structure.txt")
```

<b>Details on structure file for all black bass samples:</b> <br>
N<sub>pops</sub> = 43 <br>
N<sub>streams/lakes</sub> = 28

## ------------------------ END OF PHASE 1: GENOTYPE DATA PREPARATION -------------------------- ##

## PHASE 2: STRUCTURE ANALYSIS 
In this phase of analysis, we are assessing hierarchical population genetic structure among Spotted Bass, Northern Smallmouth Bass, and Neosho Smallmouth Bass populations in the CIH using three independent "models" in the software program STRUCTURE following a preliminary screening (Model 1) of genetic structure: Model 2) <i>Default parameters</i>, in which we use default parameter settings in STRUCTURE; Model 3) <i>Wang parameters</i>, in which we use parameter settings in STRUCTURE recommended by Wang et al. (2017); and Model 4) <i>Puechmaille metrics</i>, in which we use default parameter settings but infer population structure using metrics recommended by Puechmaille et al. (2016).

### STEP 1: Run Model 1 - Preliminary analysis; Run the code: `code/01_prelim.Rmd`
In this step, we conduct a preliminary analysis in STRUCTURE, in which we set the number of possible clusters (<i>K</i> to 28, representing all stream, lake, or hatchery systems, to screen for broad levels of population structure. We use this analysis to assess the maximum number of genetic clusters and therefore inform all downstream independent analyses.

### STEP 2: Run Model 2 - Default parameters; Run the code: `code/02_default.Rmd`
In this step, we conduct a full analysis in STRUCTURE using default parameter settings for <i>K</i> = 1-15 possible clusters (based no Model 1; see Step 1 in `01_prelim.Rmd`). We assess hierarchical structure within major genetic cluster divisions (subspecies) using <i>K</i> = 1-10 (Neosho Smallmouth Bass) and <i>K</i> = 1-5 (Northern Smallmouth Bass). 

### STEP 3: Run Model 3 - Wang parameters; Run the code: `code/03_wang.Rmd`
In this step, we conduct a full analysis in STRUCTURE using parameter settings recommended by Wang (2017) for <i>K</i> = 1-15 possible clusters (based no Model 1; see Step 1 in `01_prelim.Rmd`). We assess hierarchical structure within major genetic cluster divisions (subspecies) using <i>K</i> = 1-10 (Neosho Smallmouth Bass) and <i>K</i> = 1-5 (Northern Smallmouth Bass). 

### STEP 4: Run Model 4 - Puechmaille metrics; Run the code: `code/04_puechmaille.Rmd`
In this step, we conduct a full analysis in STRUCTURE using default parameter settings for <i>K</i> = 1-15 possible clusters (based no Model 1; see Step 1 in `01_prelim.Rmd`) but inferring population structure using metrics recommended by Puechmaille et al. (2016).

All structure output data were visualized both as q-plots (bar plots representing membership to each inferred cluster for each individual) and as pie-charts overlaid on the sampling distribution. Pie-charts were generated manually in Excel and overlaid manually onto maps generated in Analysis 1 (`01_map_analysis/smb_genetics_map_analysis.Rmd`). Pie charts represent average membership to each cluster across individuals for a given sampling site. Sampling sites correspond to Table 1 in the main text of the final ms.

## -------------- END OF PHASE 2: STRUCTURE ANALYSIS ---------------- ##

## PHASE 3: PRINCIPAL COMPONENTS ANALYSIS
In this phase of analysis, we visualize and infer population hierarchical structure at the species, subspecies, and population level using an alternative, model-free principal components analysis (PCA)

### STEP 1: Read in raw genotype data and convert to input file types for PCA
In this step, we prepare to run PCA on all genetic data by converting raw genotype data into the proper input files for 1) the R package <i>gstudio</i> (specifically for reading in allelic data in two-column format) and 2) PCA

#### 1a. Convert raw genotype data into CSV format; run the Rmd chunk below:

##### Convert genotype excel data to CSV:
```{r}
# Read in raw genotype data and convert to csv file type
genotype_data <- read_excel("../raw_data/genotype_data.xlsx") %>%
  write.csv("data/pca_data/genotype_data.csv", 
            row.names = F)
```

#### 1b. Convert genotype csv file to format readible by adegenet package; run the Rmd cunk below 

##### Convert csv to adegent file:
```{r}
# Read in and convert microsatellite genotypes to adegenet format
genotypes <- read_population("data/pca_data/genotype_data.csv", 
                             type = "column", 
                             locus.columns = 2:29)

# Save adegenet formatted genotypes
save(genotypes, file = "data/pca_data/genotypes.Rda")
```

### STEP 2: Run PCA and extract PC values.
In this step, we run PCA on allelic data to assess genetic differentiation and clustering on two hierarchical levels: 1) species level, in which both subspecies (Neosho and Northern Smallmouth Bass) and Spotted Bass are included; and 2) intraspecific groups, in which we plot samples by populations ascertained in STRUCTURE analyses. 

#### 2a. Run PCA and extract PC values for all samples combined, including Neosho Smallmouth Bass, Northern Smallmouth Bass, and Spotted Bass, to assess differentiation on the species level; run the Rmd chunk below.

##### Run PCA and extract PC values:
```{r}
# Load genotypes
load("data/pca_data/genotypes.Rda")

# Load sample metadata
load("../02_data_summary_analysis/data/metadata.Rda")

## Run PCA

# Generate multivariate allelic data table 
mv <- to_mv(genotypes, 
            drop.allele = T)

# Perform PCA
fit <- princomp(mv, 
                cor = T)

# Get proportion of variance summary for fit
summary(fit)

# Predict values from model fit
predict <- predict(fit)

# Generate data frame with predicted values
pca <- data.frame(PC1 = predict[,1],
                  PC2 = predict[,2]) 

# Filter full metadata to extract only samples with genotypes to match PCA table
metadata <- metadata %>%
  filter(genotype == "yes")

# Bind metadata and pca data for plotting
pca <- cbind(metadata, 
             pca)

# Save full pca data for plotting
save(pca, file = "data/pca_data/full_pca.Rda")
```

#### 2b. Run PCA and extract PC values for ONLY Smallmouth Bass samples, including Neosho Smallmouth Bass and Northern Smallmouth Bass, to assess differentiation on the subspecies/population level; run the Rmd chunk below.

##### Run PCA and extract PC values:
```{r}
# Load genotypes
load("data/pca_data/genotypes.Rda")

# Load sample metadata
load("../02_data_summary_analysis/data/metadata.Rda")

## Run PCA

# Filter genotypes file to include only Smallmouth Bass subspecies
genotypes <- genotypes %>%
  filter(!grepl("SPI", sample_id)) %>%
  filter(!grepl("SPE", sample_id)) %>%
  filter(!grepl("SPS", sample_id))

# Generate multivariate allelic data table 
mv <- to_mv(genotypes, 
            drop.allele = T)

# Perform PCA
fit <- princomp(mv, 
                cor = T)

# Get proportion of variance summary for fit
summary(fit)

# Predict values from model fit
predict <- predict(fit)

# Generate data frame with predicted values
pca <- data.frame(PC1 = predict[,1],
                  PC2 = predict[,2]) 

# Filter full metadata to extract only samples with genotypes to match PCA table
metadata <- metadata %>%
  filter(genotype == "yes") %>%
  filter(subspecies != "Spotted Bass")

# Bind metadata and pca data for plotting
pca <- cbind(metadata, 
             pca)

# Save full pca data for plotting
save(pca, file = "data/pca_data/subspecies_pca.Rda")
```

### STEP 3: Plot Genetic PCAs at the species and subspecies/population levels.

#### 3a. Plot Genetic PCA at the species level to assess differentiation among species and subspecies; run the Rmd chunk below.

##### Plot Genetic PCA at the species level:
```{r}
load("data/pca_data/full_pca.Rda")

# Generate genetic pca plot 
pdf("figures/species_pca.pdf", width=12, height=10)

ggplot(pca, aes(x = PC1, y = PC2, fill = subspecies)) + 
  geom_point(aes(fill = subspecies), color = "black", pch = 21, size = 3, show.legend = T, alpha = 0.9) + 
  scale_fill_manual(values = c("springgreen3", "mediumpurple", "firebrick3")) + 
  labs(fill = "Species", color = "Species", x = "PC1 (3.3 %)", y = "PC2 (2.8 %)") + 
  theme_set(theme_cowplot(12)) + 
  theme(legend.position = c(0.01,0.15)) + 
  theme(axis.text = element_text(size = 20)) + 
  theme(axis.title = element_text(size = 20)) + 
  theme(legend.text = element_text(size = 20)) + 
  theme(legend.title = element_text(size = 20, face = "bold")) + 
  theme(legend.background = element_rect(linetype = "solid", size = 0.5, color = "black")) + 
  theme(legend.margin = margin(5, 5, 5, 5)) +
  theme(panel.border = element_rect(colour = "black", fill=NA, size=1))

dev.off()
```

This plot is the basis for Figure S3a in the final ms.

#### 3b. Plot Genetic PCAs at the subspecies level to assess differentiation among species and subspecies; run the Rmd chunk below.

##### Plot Genetic PCA at the species level:
```{r}
# Load pca data at the subspecies level
load("data/pca_data/subspecies_pca.Rda")

# Generate genetic pca plot 
pdf("figures/subspecies_pca.pdf", width=12, height=10)

ggplot(pca, aes(x = PC1, y = PC2, fill = subspecies)) + 
  geom_point(aes(fill = subspecies), color = "black", pch = 21, size = 3, show.legend = T, alpha = 0.9) + 
  scale_fill_manual(values = c("springgreen3", "mediumpurple")) + 
  labs(fill = "Species", color = "Species", x = "PC1 (3.3 %)", y = "PC2 (2.8 %)") + 
  theme_set(theme_cowplot(12)) + 
  theme(legend.position = c(0.01,0.15)) + 
  theme(axis.text = element_text(size = 20)) + 
  theme(axis.title = element_text(size = 20)) + 
  theme(legend.text = element_text(size = 20)) + 
  theme(legend.title = element_text(size = 20, face = "bold")) + 
  theme(legend.background = element_rect(linetype = "solid", size = 0.5, color = "black")) + 
  theme(legend.margin = margin(5, 5, 5, 5)) +
  theme(panel.border = element_rect(colour = "black", fill=NA, size=1))

dev.off()
```

#### 3c. Plot Genetic PCA at the subspecies level to assess differentiation among populations; run the Rmd chunk below.
In this step, we used the population structure analysis output from STRUCTURE for K=9, reflecting population inference from Puechmaille (2016) metrics (see model 4 in `code/04_puechmaille.Rmd`), to assign individuals to "populations" based on majority average membership coefficients. We arbitrarily named these "populations" based on the stream systems from which individual fish were generally collected. We then generated an additional vector column in the PCA dataset ("popluation") reflecting the assigned population of each individual. Points on the PCA plot were then colored based on these assignments.

##### Plot Genetic PCA at the species level:
```{r}
# Load pca data at the subspecies level
load("data/pca_data/subspecies_pca.Rda")

# Generate new column ("population") with list of population assignments
pca$population <- c(rep("swmo", times = 244), 
                    rep("honey_syc", times = 81), 
                    rep("illinois", times = 192),
                    rep("swmo", times = 19), 
                    rep("leemulb", times = 28), 
                    rep("bayoupiney", times = 52), 
                    rep("missouri", times = 42), 
                    rep("white", times = 35), 
                    rep("skiatook", times = 20), 
                    rep("missouri", times = 20))

# Generate genetic pca plot 
pdf("figures/subspecies_pca.pdf", width=8, height=7)

ggplot(pca, aes(x = PC1, y = PC2, fill = population)) + 
  geom_point(aes(fill = population), color = "black", pch = 21, size = 3, show.legend = F, alpha = 0.9) + 
  scale_fill_manual(values = c("forestgreen", "lightblue", "darkorchid1", "gold1", "chocolate1", "maroon1", "lightpink1","blue")) + 
  labs(fill = "Species", color = "Species", x = "PC1 (3.3 %)", y = "PC2 (2.8 %)") + 
  theme_set(theme_cowplot(12)) + 
  theme(legend.position = c(0.01,0.15)) + 
  theme(axis.text = element_text(size = 20)) + 
  theme(axis.title = element_text(size = 20)) + 
  theme(legend.text = element_text(size = 20)) + 
  theme(legend.title = element_text(size = 20, face = "bold")) + 
  theme(legend.background = element_rect(linetype = "solid", size = 0.5, color = "black")) + 
  theme(legend.margin = margin(5, 5, 5, 5)) +
  theme(panel.border = element_rect(colour = "black", fill=NA, size=1))

dev.off()
```

These plots are the basis for Figure S3b-d in the final ms.

## --------------  END OF PHASE 3: PRINCIPAL COMPONENTS ANALYSIS -------------- ## 

## --------------  END OF ANALYSIS 2: HIERARCHICAL POPULATION STRUCTURE ANALYSIS --------------  ##

