---
title: "Analysis 2: Hierarchical population structure analysis"
author: "Joe Gunn"
date: "2022-07-28"
output: html_document
---

# Project: Population genetic structure and morphological differentiation between Northern Smallmouth Bass (<i>Micropterus dolomieu dolomieu</i>) and Neosho Smallmouth Bass (<i>M. d. velox</i>)
We investigated patterns of genetic (via microsatellites) and morphological diversity, differentiation, and structure across Smallmouth Bass populations (<i>Micropterus dolomieu</i>) in the Central Interior Highlands (CIH) of North America, including the two recognized subspecies: Northern Smallmouth Bass (<i>M. d. dolomieu</i>), which is native to the lower Ozark Highlands, and Neosho Smallmouth Bass (<i>M. d. velox</i>), which is endemic to tributaries of the Arkansas River Basin. We compared three independent combinations of starting parameters in the Bayesian clustering software program STRUCTURE to ascertain a robust picture of hierarchical genetic structure in the CIH. We paired these data with genetic diversity metrics (e.g., observed and expected heterozygosity and allelic richness) to determine relative amounts of variation across geographically separated populations. We then assessed differentiation in five morphometric and meristic traits. We ultimately aimed to validate or amend the taxonomic status of the Smallmouth Bass subspecies by revealing potential ecological and evolutionary divergence between them, with the hope that increased taxonomic resolution would provide insight into the presence and distribution of evolutionary significant and management units for a popular sportfish.

## Specific Aim: Hierarchical structure analysis with three independent methods
For this aim, we use Bayesian clustering analysis and genotypes from 14 microsatellite loci with STRUCTURE (see citation below in under "Programs Needed") to assess patterns hierarchical genetic structure among and within black bass species (i.e., Spotted Bass and Smallmouth Bass) and subspecies (i.e., Northern Smallmouth Bass and Neosho Smallmouth Bass) in the CIH. To maximize our ability to ascertain potentially complex or cryptic levels of genetic diversity, we employ and integrate three independent "methods" within STRUCTURE, which each use either different starting parameters for identifying genetic clusters or different inference techniques for determining the optimal number of clusters: 1) <i>default parameters</i> (default parameters in STRUCUTRE); 2) <i>Wang parameters</i> (parameters recommended by Wang (2017); and 3) <i>Puechmaille metrics</i> (optimal cluster inference method recommended by Puechmaille 2016).

## Phases of Analysis
### Phase 1: Genotype data preparation
### Phase 2: Structure analysis with default STRUCTURE parameters
### Phase 3: Structure analysis with Wang (2017) recommended STRUCTURE parameters
### Phase 4: Structure analysis with Puechmaille metrics inference 
### Phase 5: Principal Components Analysis

Programs needed:

STRUCTURE v.2.3.4. (Pritchard et al. 2000)

Citation:

Pritchard JK, Stephens M, Donnelly P. 2000. Inference of population structure using multilocus genotype data. Genetics 155: 945-959.

CLUMPP v.1.1.2 (Jakobsson and Rosenberg 2007)

Citation:

Jakobsson M, Rosenberg NA. 2007. CLUMPP: A cluster matching and permutation program for dealing with label switching and multimodality in analysis of population structure. Bioinformatics 23: 1801-1806.

We ran each hierarchical analysis in separate .Rmd files, which are found in the `code/` directory.

## Libraries needed for analysis
```{r}
library(tidyverse)
library(cowplot)
library(readxl)
library(writexl)
library(genepopedit)
<<<<<<< HEAD
=======
library(graph4lg)
>>>>>>> 821a6d0226fa418c2c3a582ae1f2eba73e538796
```

## PHASE 1: GENOTYPE DATA PREPARATION
In this phase of the analysis, we convert the raw microsatellite genotype data (`../raw_data/genotype_data.xlsx`) into Genepop-compatible format, which is read by the software program STRUCTURE in all downstream analyses.

### STEP 1: Manually reformat Excel genotype data into genepop-compatible format.
In this step, we opted to reformat the raw Excel microsatellite genotype data by hand in Excel, given that we could not find a working R library and/or function that converts raw genotype tables into genepop format (i.e., we attempted to use the <i>genind_to_genepop()</i> function from the package <i>graph4lg</i>, but we could not get our genind to convert properly). We follwed the steps below exactly in Excel to convert our genotype data into Genepop format, and we saved the corresponding data as a text file in the working `data` directory.

#### 1a. Add a "," after each sample_id in the first column.

#### 1b. Insert a row above each new population in the data, and insert "pop" in the sample_id column in each empty row (the word "pop" should appear for each population)

##### 1c. Copy the header row of microsatellite locus IDs and transpose paste in a new Excel sheet; each locus ID should be listed in its own row (number of rows should be equal to numbrer of locus IDs)

##### 1d. Copy all genotype and label data from the original Excel file and paste in the row immediately below the list of locus IDs

##### 1e. Save the Excel sheet as a .xlsx and a .txt file, with a specification that it is a "genepop" file.

### STEP 2: Generate STRUCTURE formatted files from genepop format.
<<<<<<< HEAD
In this step, we are reading in, checking, and converting the genepop file manually generated in STEP 1 above into a STRUCTURE formatted file. There are quite a few tricky and very specific formatting needs for the genepop text file before genepopedit will convert successfully. Follow Step 2a-2f closely to create a proper genepop format. Click on the text file manually here, in R Studio, and use Ctrl + F in R Studio to find and replace the necessary things.
=======
In this step, we are reading in, checking, and converting the genepop file manually generated in STEP 1 above into a STRUCTURE formatted file. There are quite a few tricky and very specific formatting needs for the genepop text file before genepopedit will convert successfully. Follow Step 2a-... closely to create a proper genepop format. Click on the text file manually here, in R Studio, and use Ctrl + F in R Studio to find and replace the necessary things.
>>>>>>> 821a6d0226fa418c2c3a582ae1f2eba73e538796

### IMPORTANT NOTE ON DATA: genepopedit requires that sample names be in the format 'popname_01', 'popname_02', etc. where the 'popname' is a signifier for the population of origin for the sample, and the '01', etc. are numerical designations for the samples. 'popname' and the numeric must be separated by a "_", and there can only be a single "_". Otherwise, genepopedit will not read the sample's population of origin correctly. In the case of this project, we are clumping samples together by stream for STRUCTURE analysis to see if there is substructure mapped by stream. Many sample IDs in our dataset were not in the required format, so we added an additional column to the metadata set and associated processed data sets called "structure_ID", which gives a unique sample name in the proper structure format. This column can then be linked back to any other column in downstream analyses if we need to know which specific samples the results came from. 

#### 2a. All quotation marks were removed from the text file

#### 2b. All "," in the genepop text files were manually replaced with " ,  "

#### 2c. All "tabs" between six-digit alleles were were manually replaced with " " (a single space)

#### 2d. The top row designating the dataset combination was manually omitted

#### 2e. Any additional space at the bottom of the text file was removed

#### 2f. Generate STRUCTURE file from genepop file
Here it is important to not include the 'popgroup = ' unless it is explicitly necessary. Assigning popgroup to a dataframe with population names will cause the STRUCTURE input file to have word strings in the popdata column. Our experience is that STRUCTURE expects an integer here rather than a string, so best not to use popgroup.

##### Convert genepop to STRUCTURE format:`
```{r}
# Generate structure file
genepop_structure("data/genepop.txt", 
                  locusnames = TRUE, 
                  path = "data/structure.txt")
```

<b>Details on structure populations for all black bass samples:</b> <br>
N<sub>pops</sub> = 43

## ------------------------ END OF PHASE 1: GENOTYPE DATA PREPARATION -------------------------- ##

## PHASE 2: STRUCTURE ANALYSIS WITH DEFAULT STRUCTURE PARAMETERS

<<<<<<< HEAD
#### Run the code: `code/02_default.Rmd`

## ------------------------ END OF PHASE 2: STRUCTURE ANALYSIS WITH DEFAULT STRUCTURE PARAMETERS -------------------------- ##

## PHASE 3: STRUCTURE ANALYSIS WITH WANG (2017) RECOMMENDED STRUCTURE PARAMETERS: `code/03_wang.Rmd`
=======
#### Run the code: `code/01_default.Rmd`

## ------------------------ END OF PHASE 2: STRUCTURE ANALYSIS WITH DEFAULT STRUCTURE PARAMETERS -------------------------- ##

## PHASE 3: STRUCTURE ANALYSIS WITH WANG (2017) RECOMMENDED STRUCTURE PARAMETERS: `code/02_wang.Rmd`
>>>>>>> 821a6d0226fa418c2c3a582ae1f2eba73e538796
In this phase of analysis, we are assessing population structure 

## ------------------------ END OF PHASE 3: STRUCTURE ANALYSIS WITH WANG (2017) RECOMMENDED STRUCTURE PARAMETERS ---------- ##

<<<<<<< HEAD
## PHASE 4: STRUCTURE ANALYSIS WITH PUECHMAILLE METRICS: `code/04_puechmaille.Rmd`
=======
## PHASE 4: STRUCTURE ANALYSIS WITH PUECHMAILLE METRICS: `code/03_puechmaille.Rmd`
>>>>>>> 821a6d0226fa418c2c3a582ae1f2eba73e538796
In this phase of analysis, we are assessing population structure 

## ------------------------ END OF PHASE 4: STRUCTURE AND HYBRID ASSIGNMENT ANALYSIS OF SMB AND CIH ----------- ##

## PHASE 5: PRINCIPAL COMPONENTS ANALYSIS
In this phase of analysis, we visualize and infer population hierarchical structure at the species, subspecies, and population level using an alternative, model-free principal components analysis (PCA)

## ------------------------ END OF PHASE 5: PRINCIPAL COMPONENTS ANALYSIS ----------- ##

## ------------------------ END OF ANALYSIS 2: HIERARCHICAL POPULATION STRUCTURE ANALYSIS ----------------------- ##

