---
title: "Analysis 2: Phase 2: Model 4 - Structure analysis using Puechmaille inference metrics"
author: "Joe Gunn"
date: "2022-08-13"
output: html_document
---

# Project: Population genetic structure and morphological differentiation between Northern Smallmouth Bass (<i>Micropterus dolomieu dolomieu</i>) and Neosho Smallmouth Bass (<i>M. d. velox</i>)
We investigated patterns of genetic (via microsatellites) and morphological diversity, differentiation, and structure across Smallmouth Bass populations (<i>Micropterus dolomieu</i>) in the Central Interior Highlands (CIH) of North America, including the two recognized subspecies: Northern Smallmouth Bass (<i>M. d. dolomieu</i>), which is native to the lower Ozark Highlands, and Neosho Smallmouth Bass (<i>M. d. velox</i>), which is endemic to tributaries of the Arkansas River Basin. We compared three independent combinations of starting parameters in the Bayesian clustering software program STRUCTURE to ascertain a robust picture of hierarchical genetic structure in the CIH. We paired these data with genetic diversity metrics (e.g., observed and expected heterozygosity and allelic richness) to determine relative amounts of variation across geographically separated populations. We then assessed differentiation in five morphometric and meristic traits. We ultimately aimed to validate or amend the taxonomic status of the Smallmouth Bass subspecies by revealing potential ecological and evolutionary divergence between them, with the hope that increased taxonomic resolution would provide insight into the presence and distribution of evolutionary significant and management units for a popular sportfish.

## Specific Aim: Hierarchical structure analysis with three independent methods
For this aim, we use Bayesian clustering analysis and genotypes from 14 microsatellite loci with STRUCTURE (see citation below in under "Programs Needed") to assess patterns hierarchical genetic structure among and within black bass species (i.e., Spotted Bass and Smallmouth Bass) and subspecies (i.e., Northern Smallmouth Bass and Neosho Smallmouth Bass) in the CIH. To maximize our ability to ascertain potentially complex or cryptic levels of genetic diversity, we employ and integrate three independent "methods" within STRUCTURE, which each use either different starting parameters for identifying genetic clusters or different inference techniques for determining the optimal number of clusters: 1) <i>default parameters</i> (default parameters in STRUCUTRE); 2) <i>Wang parameters</i> (parameters recommended by Wang (2017); and 3) <i>Puechmaille metrics</i> (optimal cluster inference method recommended by Puechmaille 2016).

### Libraries needed for analysis
```{r}
library(tidyverse)
library(cowplot)
library(readxl)
library(writexl)
library(pophelper)
```

## PHASE 2: STRUCTURE ANALYSIS 
In this phase of analysis, we are assessing hierarchical population genetic structure among Spotted Bass, Northern Smallmouth Bass, and Neosho Smallmouth Bass populations in the CIH using three independent "models" in the software program STRUCTURE following a preliminary screening (Model 1) of genetic structure: Model 2) <i>Default parameters</i>, in which we use default parameter settings in STRUCTURE; Model 3) <i>Wang parameters</i>, in which we use parameter settings in STRUCTURE recommended by Wang (2017); and Model 4) <i>Puechmaille metrics</i>, in which we use default parameter settings but infer population structure using metrics recommended by Puechmaille et al. (2016).

    Important note: We originally conducted population structure analyses on a PC Desktop computer using the STRUCTURE GUI, and we did not set a seed value for each of our analyses. Thus, when we re-ran STRUCTURE for each of our "models", we were unable to reproduce the exact output values from the analyses, including deltaK and all other parameters. However, we verified that all inferences relative to the new output values (i.e., the number of genetic clusters <i>K</i> and patterns of individual membership coefficients) were matched to the original inferences in the Gunn et al. (2020) publication.
    
### STEP 4: Run Model 4 - Puechmaille metrics
In this step, we conduct a full analysis in STRUCTURE using default parameter settings for <i>K</i> = 1-15 possible clusters (based no Model 1; see Step 1 in `01_prelim.Rmd`) but inferring population structure using metrics recommended by Puechmaille et al. (2016). 

## PHASE 4: STRUCTURE ANALYSIS WITH PUECHMAILLE INFERENCE METRICS
In this phase of analysis, we are assessing hierarchical population genetic structure among Spotted Bass, Northern Smallmouth Bass, and Neosho Smallmouth Bass populations in the CIH using default STRUCTURE parameter settings, but inferring optimal number of genetic clusters using inference metrics developed by Puechmaille (2017).

### STEP 4: Run Model 2 - Default paramters
In this step, we conduct a full analysis in STRUCTURE using default parameter settings for <i>K</i> = 1-15 possible clusters (based no Model 1; see Step 1 in `01_prelim.Rmd`). We do this in two analyses: Analysis 1) all samples together, and Analysis 2) samples broken down into the major divisions in the data based on the results (to assess the degree of substructure within major clusters). 

#### 4a: Generate batch list files and shell scripts for running STRUCTURE in parallel
In this step, we are creating batch lists of command line code to run STRUCTURE analyses in parallel. Each batch list contains a separate line of code to run a single replicate at an a priori determined number of populations (<i>K</i>). We run the analysis for 10 replicates at each <i>K</i>, e.g.:

structure -K 1 -m mainparams -e extraparams -i structure_input.txt -o structure_output_1_1
structure -K 1 -m mainparams -e extraparams -i structure_input.txt -o structure_output_1_2
structure -K 1 -m mainparams -e extraparams -i structure_input.txt -o structure_output_1_3
structure -K 1 -m mainparams -e extraparams -i structure_input.txt -o structure_output_1_4
structure -K 1 -m mainparams -e extraparams -i structure_input.txt -o structure_output_1_5
.
.
.
structure -K 1 -m mainparams -e extraparams -i structure_input.txt -o structure_output_1_10
structure -K 2 -m mainparams -e extraparams -i structure_input.txt -o structure_output_2_1
structure -K 2 -m mainparams -e extraparams -i structure_input.txt -o structure_output_2_2
.
.
.
where, in the output name, the first number represents the <i>K</i> value for the run, and the last number represents the replicate.

##### 4a.1 Generate batch list file; run the Rmd chunk below:
The Rmd chunk below generates a list of commands that contains the full cluster paths (/home/data/...) for necessary files.

##### Generate batch commands: `batch_cmd_lists/default_batch_cmd_list.txt`
```{r}
nk <- data.frame(c(1:15))
nreps <- data.frame(c(1:10))

cat("", file="batch_cmd_lists/default_batch_cmd_list.txt")

# Run loop to create file for storing commands
for(ii in 1:nrow(nk)) {

  for(aa in 1:nrow(nreps)) {
    
    structure_call <- paste("structure ")

    param_files <- paste(" -m /home/jcg5g9/data/SMB_Genetics/03_structure_analysis/data/structure_data/param_files/mainparams_default -e /home/jcg5g9/data/SMB_Genetics/03_structure_analysis/data/structure_data/param_files/extraparams_default")

    input <- paste(" -i /home/jcg5g9/data/SMB_Genetics/03_structure_analysis/data/structure_data/input_data/structure.txt")
  
    output <- paste(" -o /home/jcg5g9/data/SMB_Genetics/03_structure_analysis/data/structure_data/output_data/default/full_data/")
  
    cat(paste(structure_call, "-K ", nk[ii,], param_files, input, output,
            strsplit("default_structure.txt", "_structure.txt", fixed = TRUE)[[1]][1], "_", nk[ii,], "_", nreps[aa,], sep=""),
      "\n", 
      file = paste("batch_cmd_lists/default_batch_cmd_list.txt"),
      append = TRUE)
  }
}
```

##### 4a.2. Generate shell script; run the Rmd chunk below:

##### IMPORTANT NOTE: This shell script was generated on the server directly, and thus the Rmd chunk below does not need to be run to generate the file. This is purely to keep a record of each script file. 

##### Generate batch commands: `shell_scripts/default_structure.sh`
```{r}
#!/bin/bash
#-------------------------------------------------------------------------------
#  SBATCH CONFIG
#-------------------------------------------------------------------------------
## resources
#SBATCH --partition Lewis
#SBATCH --nodes=1
#SBATCH --ntasks=1  # used for MP#SBATCH -e error_%A_%a.err # Standard error codes, otherwise leav$
##SBATCH --cpus-per-task=12  # cores per task
#SBATCH --mem-per-cpu=16G  # memory per core (default is 1GB/core)
#SBATCH --time 2-00:00  # days-hours:minutes
#SBATCH --qos=normal
#SBATCH --array=1-150

## labels and outputs
#SBATCH --job-name=smb_genetics_structure_jgunn
#
#SBATCH -o test_%A_%a.out # Standard output
#SBATCH -e error_%A_%a.err # Standard error

## notifications
#SBATCH --mail-user=jcg5g9@mail.missouri.edu  # email address for notifications
#SBATCH --mail-type=END,FAIL  # which type of notifications to send
#-------------------------------------------------------------------------------

#echo "### Starting at: $(date) ###"


# load packages
#module load rss/rss-2020
#module load structure/structure-2.3.4

#COMMANDA=`head -n ${SLURM_ARRAY_TASK_ID} ../batch_cmd_lists/default_batch_cmd_list.txt | tail -n 1`
#eval $COMMANDA

#echo "### Ending at: $(date) ###"
```

#### 4b. Prepare mainparam and extraparam files for STRUCTURE.
In this step, we are preparing the mainparam and extraparam input files for STRUCTURE so that they are unique to each analysis we are running.

##### 4b.1. Edit base mainparam STRUCTURE file (downloaded with STRUCTURE program) and generate a separate, unique mainparam file.

<b>mainparam file:</b> <br>
`../data/structure_data/param_files/mainparams_default` <br> 

###### 4b.1.1. Edit "maxpops" value to reflect the number of populations designated in Step 3b above. These values should be the following:

<b>Populations:</b> 15 <br>

###### 4b.1.2. Set the number of burn-in and MCMC iterations to run; these are the same for each analysis:

<b>Burn-in runs:</b> 500,000 <br>
<b>MCMC runs:</b> 1,000,000 <br>

###### 4b.1.3. Set the number of individuals.

<b>Individuals:</b> 766 <br>

###### 4b.1.4 Set the number of loci.

<b>Loci:</b> 14 <br>

###### 4b.1.5. Set 'ONEROWPERIND' to '0'

###### 4b.1.6. Set 'LABEL' to '1'

###### 4b.1.7. Set 'POPDATA' to '1'

###### 4b.1.8. Set 'POPFLAG' to '0'

##### 4b.2. Edit base extraparam STRUCTURE file (downloaded with STRUCTURE program).
We did not change any of the default settings in the extraparams file (most importantly, we used the default admixture model), so we only used a single extraparams file.

<b>extraparam file:</b> <br>
`../data/structure_data/param_files/extraparams_default` <br> 

#### 4c. Population Structure Analysis with STRUCTURE

See `smb_genetics_structure_analysis.Rmd` (Line 19) for programs needed for analysis.

##### 4c.1. Run STRUCTURE analysis using the input data generated in PHASE 1 (`smb_genetics_structure_analysis.Rmd`). Navigate to `shell_scripts/` Be sure that all relative and full paths to all input files and output destination directories are set up properly (ideally, this is already done within this GitHub repo). This command line code assumes capability to run the code using SLURM or a SLURM-like cluster scheduling software.

Run `sbatch default_structure.sh`

##### 4c.2. Structure output files are generated and stored here: `../data/structure_data/output_data/default/`

##### 4c.3. Compress output directory into a zip file compatible with Structure Selector (Li and Liu 2017) or Structure Harvester (Earl and vonHoldt 2011) online

##### 4c.4. Submit zip directory to Structure Selector or Structure Harvester to extract summary results; analyze results using Puechmaille metrics (MedMedK, MedMeanK, MaxMedK, MaxMeanK) using a threshold value of 0.5
We analyzed STRUCTURE results using the four metrics recommended by Puechmaille (2016).

<b>Model results</b>:
We found strongest evidence for K=9 major clusters; all metrics were optimzed at 9 unique clusters.

##### 4c.5. Visualize STRUCTURE runs for all data combinations

##### 4c.6. Convert STRUCTURE files into aligned Q files compatible with analysis in the program CLUMPP (Jakobbson and Rosenberg 2007); run the rmd chunk below

##### Convert STRUCTURE files to aligned Q files for CLUMPP: 
```{r}
# Get a list of structure files for each data combination
sfiles <- list.files("../data/structure_data/output_data/default/full_data/",
                     full.names = T)

# Extract q value information (ancestry proportions) from each run for each K value for each individual from the STRUCTURE output files in the directories listed above
Q <- readQ(sfiles)

# Tabulate information from the q lists
tab <- tabulateQ(Q)

# Summarize information from tabulations above
summary <- summariseQ(tab)

# Extract deltaK and associated summary information using Evanno method
evanno <- evannoMethodStructure(summary, 
                                returnplot = F)

# Set infinity and NA to zero arbitrarily
evanno$deltaK[evanno$deltaK == "Inf"] <- 0
evanno$deltaK[is.na(evanno$deltaK)] <- 0

# Write Evanno table to Excel table for manuscript preparation.

## Convert to data frame
evanno <- as.data.frame(evanno)

## save evanno file for downstream analyses
save(evanno, file = "../data/structure_data/deltak_data/puechmaille_deltak.Rda")

# Align replicate runs for each K to correct label switching
align <- alignK(Q)

# Save aligned replicates file for downstream analyses
save(align, file = "../data/structure_data/aligned_data/puechmaille_align.Rda")
```

##### 4c.7. Export CLUMPP compatible files for CLUMPP analysis; run the rmd chunk below to export CLUMPP associated files for each K.

<b>We used the following parameters: </b> <br>

Large-K-Greedy algorithm (paramrep = 3) <br>
10,000 replicates <br>

### IMPORTANT NOTE: This step only needs to be run ONCE to generate files for CLUMPP. Once you have run this chunk, move on to Step 2e.3. Uncomment each line to run this code.

##### Export CLUMPP files:
```{r, echo = FALSE}
# Load aligned structure replicate data
load("../data/structure_data/aligned_data/puechmaille_align.Rda")

# Generate clumpp formatted file
clumppExport(align, 
            parammode = 3, 
            paramrep = 10000,
            exportpath = "../data/structure_data/clumpp_data/puechmaille_clumpp/")
```

The code above generates a .txt file (pop_K#-combined.txt) with combined cluster ancestry proportions for each individual at each K and stores it in a separate directory labeled "pop_K#", where '#' is the corresponding K value. Additionally, the code generates an accompanying "paramfile" that is input for CLUMPP, which contains information on parameters to generate merged datasets across our 10 replicates at each K.

##### 4c.8. Generate merged Q files using CLUMPP.
In this step, we used the software program CLUMPP (Jakobbson and Rosenberg 2007) to merge cluster ancestry proportion inferences for all replicate runs of each K across individuals. We downloaded the Linux 64-bit build of CLUMPP v.1.1.2 and installed it on our computing cluster. We then executed the program for each analysis by sequentially copying the associated paramfiles generated in the clumppExport function (see section above) to the CLUMPP home directory. All output files were then moved to the corresponding structure directory.

###### 4c.8.1. Downolad CLUMPP and install on cluster; navigate to the desired directory: `code`. Run the command: `wget https://rosenberglab.stanford.edu/software/CLUMPP_Linux64.1.1.2.tar.gz`

###### 4c.8.2. Run the command: `gunzip CLUMPP_Linux64.1.1.2.tar.gz`

###### 4c.8.3. Run the command: `tar -xvf CLUMPP_Linux64.1.1.2.tar`
cd 
###### 4c.8.4. For each analysis separately, copy the 'pop_K#-combined.txt' file and 'paramfile' over to the CLUMPP directory (`CLUMPP_Linux64.1.1.2`)

###### 4c.8.5. For each analysis separately, execute CLUMPP on the paramfile; run the code `./CLUMPP paramfile`
We are running this code with GREEDY OPTION 2, which uses random sampling of replicate runs. 

This code generates three additional files, which we moved back to the corresponding pop_K folder in the clumpp_data directory:

<b>Files generated: </b> <br>

pop_K#-combined-miscfile.txt <br>
pop_K#-combined-merged.txt <br>
pop_K#-combined-aligned.txt <br>

#### 4d. Plot Puechmaille results and STRUCTURE runs at best K value (according to Puechmaille metrics) for each analysis
Here we plot the results of structure analysis using default parameters and the Puechmaille method of cluster inference. The Puechmaille method revealed greatest relative support to <i>K</i>=9 clusters.

Rather than generating a plot of Puechmaille metrics, we opted to only generate a structure q-plot showing cluster membership of each individual at <i>K</i>=9 clusters

##### 4d.1. Read in sample metadata file and filter for samples with genotypes; run the Rmd chunk below.
In this step, we read in the full sample metadata and filter for only those samples used in genetic analyses (to be used for plotting)

##### Read in and filter metadata:
```{r}
# Load sample metadata
load("../../02_data_summary_analysis/data/metadata.Rda") 

# Filter metadata for only those samples with genotype data and convert factors to characters
metadata <- metadata %>%
  filter(genotype == "yes") %>%
  mutate(sample_id = as.character(sample_id), 
         population = as.character(population))

# save metadata for plotting
save(metadata, file = "../data/structure_data/metadata/full_default_metadata.Rda")
```

##### 4d.2. Plot STRUCTURE results for Puechmaille metrics; run the Rmd chunk below.

##### STRUCTURE results for default parameters: `puechmaille_k9.pdf`
```{r}
# Load full default metadata
load("../data/structure_data/metadata/full_default_metadata.Rda")

# Read in combined-merged Q table from CLUMPP for K=2
default_k9 <- readQ("../data/structure_data/clumpp_data/puechmaille_clumpp/pop_K9/pop_K9-combined-merged.txt")

# Plot structure results
plotQ(default_k9,
      grplab = metadata[,c(1,4)],
      ordergrp = F,
      selgrp = "population",
      showgrplab = T,
      showlegend = F, 
      showsp = F, 
      showdiv = T, 
      divsize = 2, 
      divtype = 1,
      divcol = "black",  
      grplabsize = 4, 
      grplabangle = 45,
      grplabjust = 0.5,
      legendkeysize = 15, 
      linealpha = 0,
      legendtextsize = 10, 
      linesize = 0.05, 
      pointsize = 4, 
      barbordercolour = "black",
      barbordersize = 0.1, 
      clustercol = c("firebrick3", "lightskyblue", "darkorchid1", "blue", "chocolate1", "forestgreen", "goldenrod2", "lightpink1","mediumpurple"),
      exportplot = T,
      imgtype = "pdf",
      height = 10, 
      width = 100,
      outputfilename = "puechmaille_k9",
      exportpath = "../figures")
```

## ------------------------ END OF PHASE 2: STRUCTURE ANALYSIS ----------------------- ##
